--- squid-3.3.9/helpers/basic_auth/DB/config.test~	2013-09-11 06:08:38.000000000 +0200
+++ squid-3.3.9/helpers/basic_auth/DB/config.test	2013-10-23 13:13:59.831569931 +0200
@@ -2,6 +2,6 @@
 
 ## Test: do we have perl to build the helper scripts?
 ## Test: do we have pod2man to build the manual?
-perl --version >/dev/null && echo | pod2man >/dev/null
+perl --version >/dev/null && echo =pod | pod2man >/dev/null
 
 exit $?
--- squid-3.3.9/helpers/external_acl/SQL_session/config.test~	2013-09-11 06:08:38.000000000 +0200
+++ squid-3.3.9/helpers/external_acl/SQL_session/config.test	2013-10-23 13:15:36.359232506 +0200
@@ -2,6 +2,6 @@
 
 ## Test: do we have perl to build the helper scripts?
 ## Test: do we have pod2man to build the manual?
-perl --version >/dev/null && echo | pod2man >/dev/null
+perl --version >/dev/null && echo =pod | pod2man >/dev/null
 
 exit $?
--- squid-3.3.9/helpers/log_daemon/DB/config.test~	2013-09-11 06:08:38.000000000 +0200
+++ squid-3.3.9/helpers/log_daemon/DB/config.test	2013-10-23 13:16:00.016143347 +0200
@@ -2,6 +2,6 @@
 
 ## Test: do we have perl to build the helper scripts?
 ## Test: do we have pod2man to build the manual?
-perl --version >/dev/null && echo | pod2man >/dev/null
+perl --version >/dev/null && echo =pod | pod2man >/dev/null
 
 exit $?
