Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.39 squid/src/cf.data.pre:1.245.2.40
*** squid/src/cf.data.pre:1.245.2.39	Tue Apr  1 21:39:02 2003
--- squid/src/cf.data.pre	Mon Apr 14 13:55:39 2003
***************
*** 3007,3012 ****
--- 3007,3013 ----
  
  		header_access Allow allow all
  		header_access Authorization allow all
+ 		header_access WWW-Authenticate allow all
  		header_access Cache-Control allow all
  		header_access Content-Encoding allow all
  		header_access Content-Length allow all
