Index: squid/src/ftp.c
diff -c squid/src/ftp.c:1.316.2.29 squid/src/ftp.c:1.316.2.30
*** squid/src/ftp.c:1.316.2.29	Thu Sep  1 16:14:45 2005
--- squid/src/ftp.c	Thu Sep  1 16:28:29 2005
***************
*** 2279,2285 ****
  {
      int code = ftpState->ctrl.replycode;
      debug(9, 3) ("This is ftpReadTransferDone\n");
!     if (code == 226) {
  	/* Connection closed; retrieval done. */
  	if (ftpState->flags.html_header_sent)
  	    ftpListingFinish(ftpState);
--- 2279,2285 ----
  {
      int code = ftpState->ctrl.replycode;
      debug(9, 3) ("This is ftpReadTransferDone\n");
!     if (code == 226 || code == 250) {
  	/* Connection closed; retrieval done. */
  	if (ftpState->flags.html_header_sent)
  	    ftpListingFinish(ftpState);
***************
*** 2345,2351 ****
  {
      int code = ftpState->ctrl.replycode;
      debug(9, 3) ("This is ftpWriteTransferDone\n");
!     if (code != 226) {
  	debug(9, 1) ("ftpReadTransferDone: Got code %d after sending data\n",
  	    code);
  	ftpFailed(ftpState, ERR_FTP_PUT_ERROR);
--- 2345,2351 ----
  {
      int code = ftpState->ctrl.replycode;
      debug(9, 3) ("This is ftpWriteTransferDone\n");
!     if (!(code == 226 || code == 250)) {
  	debug(9, 1) ("ftpReadTransferDone: Got code %d after sending data\n",
  	    code);
  	ftpFailed(ftpState, ERR_FTP_PUT_ERROR);
***************
*** 2521,2527 ****
  	storeUrl(ftpState->entry), code);
      if (cbdataValid(ftpState))
  	debug(9, 5) ("ftpSendReply: ftpState (%p) is valid!\n", ftpState);
!     if (code == 226) {
  	err_code = (ftpState->mdtm > 0) ? ERR_FTP_PUT_MODIFIED : ERR_FTP_PUT_CREATED;
  	http_code = (ftpState->mdtm > 0) ? HTTP_ACCEPTED : HTTP_CREATED;
      } else if (code == 227) {
--- 2521,2527 ----
  	storeUrl(ftpState->entry), code);
      if (cbdataValid(ftpState))
  	debug(9, 5) ("ftpSendReply: ftpState (%p) is valid!\n", ftpState);
!     if (code == 226 || code == 250) {
  	err_code = (ftpState->mdtm > 0) ? ERR_FTP_PUT_MODIFIED : ERR_FTP_PUT_CREATED;
  	http_code = (ftpState->mdtm > 0) ? HTTP_ACCEPTED : HTTP_CREATED;
      } else if (code == 227) {
