Index: squid3/src/external_acl.cc
diff -c squid3/src/external_acl.cc:1.54 squid3/src/external_acl.cc:1.55
*** squid3/src/external_acl.cc:1.54	Mon Oct 20 06:33:01 2003
--- squid3/src/external_acl.cc	Wed Nov 19 10:18:35 2003
***************
*** 586,592 ****
  
          key = makeExternalAclKey(ch, acl);
  
!         ch->auth_user_request = NULL;
  
          if (!key) {
              /* Not sufficient data to process */
--- 586,593 ----
  
          key = makeExternalAclKey(ch, acl);
  
!         if (acl->def->require_auth)
!             ch->auth_user_request = NULL;
  
          if (!key) {
              /* Not sufficient data to process */
