2001-09-09 21:59  Henrik Nordstrom <hno@squid-cache.org>

        statHistDump prototype mismatch

Index: squid/src/protos.h
diff -u squid/src/protos.h:1.387.2.6 squid/src/protos.h:1.387.2.7
--- squid/src/protos.h:1.387.2.6	Sat May 19 18:09:59 2001
+++ squid/src/protos.h	Sun Sep  9 13:59:28 2001
@@ -745,7 +745,7 @@
 extern void statHistCopy(StatHist * Dest, const StatHist * Orig);
 extern void statHistSafeCopy(StatHist * Dest, const StatHist * Orig);
 extern double statHistDeltaMedian(const StatHist * A, const StatHist * B);
-extern void statHistDump(const StatHist * H, StoreEntry * sentry, StatHistBinDumper bd);
+extern void statHistDump(const StatHist * H, StoreEntry * sentry, StatHistBinDumper * bd);
 extern void statHistLogInit(StatHist * H, int capacity, double min, double max);
 extern void statHistEnumInit(StatHist * H, int last_enum);
 extern void statHistIntInit(StatHist * H, int n);
