diff -urN squid-3.0.STABLE9.orig/configure.in squid-3.0.STABLE9/configure.in
--- squid-3.0.STABLE9.orig/configure.in	2008-09-09 18:06:45.000000000 +0200
+++ squid-3.0.STABLE9/configure.in	2008-09-19 07:43:53.042671496 +0200
@@ -36,6 +36,7 @@
 
 
 CRYPTLIB=''
+NSLLIB=''
 REGEXLIB=''	# -lregex
 LIBREGEX=''	# libregex.a
 
@@ -2205,8 +2206,10 @@
 #include <sys/msg.h>])
 
 dnl Check for needed libraries
-AC_CHECK_LIB(nsl, main)
-AC_CHECK_LIB(socket, main)
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname))
+AC_CHECK_LIB(nsl, yp_match, NSLLIB="-lnsl")
+AC_SUBST(NSLLIB)
+AC_SEARCH_LIBS(socket, socket)
 dnl Check for Winsock only on MinGW, on Cygwin we must use emulated BSD socket API
 case "$host_os" in
   mingw|mingw32)
diff -urN squid-3.0.STABLE9.orig/helpers/basic_auth/YP/Makefile.am squid-3.0.STABLE9/helpers/basic_auth/YP/Makefile.am
--- squid-3.0.STABLE9.orig/helpers/basic_auth/YP/Makefile.am	2008-09-09 18:06:40.000000000 +0200
+++ squid-3.0.STABLE9/helpers/basic_auth/YP/Makefile.am	2008-09-19 07:44:22.396001181 +0200
@@ -7,5 +7,5 @@
 
 libexec_PROGRAMS = yp_auth
 yp_auth_SOURCES	= yp_auth.c nis_support.h nis_support.c
-LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(XTRA_LIBS)
+LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(NSLLIB) $(XTRA_LIBS)
 INCLUDES	= -I$(top_srcdir)/include
