Index: squid/src/main.c
diff -c squid/src/main.c:1.345.2.10 squid/src/main.c:1.345.2.11
*** squid/src/main.c:1.345.2.10	Mon May  5 18:24:14 2003
--- squid/src/main.c	Sun Jun  8 17:28:46 2003
***************
*** 881,886 ****
--- 881,888 ----
       */
      /* Connect stdio to /dev/null in daemon mode */
      nullfd = open("/dev/null", O_RDWR | O_TEXT);
+     if (nullfd < 0)
+ 	fatalf("/dev/null: %s\n", xstrerror());
      dup2(nullfd, 0);
      if (opt_debug_stderr < 0) {
  	dup2(nullfd, 1);
