Index: squid/src/http.c
diff -c squid/src/http.c:1.384.2.28 squid/src/http.c:1.384.2.29
*** squid/src/http.c:1.384.2.28	Fri Feb 11 03:52:59 2005
--- squid/src/http.c	Wed Mar  9 07:40:04 2005
***************
*** 744,750 ****
  		    if ((len < 0 && !ignoreErrno(errno)) || len == 0) {
  			keep_alive = 0;
  		    } else if (len > 0) {
! 			debug(11, 1) ("httpReadReply: Excess data from \"%s %s\"\n",
  			    RequestMethodStr[httpState->orig_request->method],
  			    storeUrl(entry));
  			storeAppend(entry, buf, len);
--- 744,751 ----
  		    if ((len < 0 && !ignoreErrno(errno)) || len == 0) {
  			keep_alive = 0;
  		    } else if (len > 0) {
! 			debug(11, Config.onoff.relaxed_header_parser <= 0 || keep_alive ? 1 : 2)
! 			    ("httpReadReply: Excess data from \"%s %s\"\n",
  			    RequestMethodStr[httpState->orig_request->method],
  			    storeUrl(entry));
  			storeAppend(entry, buf, len);
***************
*** 782,788 ****
  	    return;
  	case -1:
  	    /* Server is nasty on us. Shut down */
! 	    debug(11, 1) ("httpReadReply: Excess data from \"%s %s\"\n",
  		RequestMethodStr[httpState->orig_request->method],
  		storeUrl(entry));
  	    fwdComplete(httpState->fwd);
--- 783,790 ----
  	    return;
  	case -1:
  	    /* Server is nasty on us. Shut down */
! 	    debug(11, Config.onoff.relaxed_header_parser <= 0 || entry->mem_obj->reply->keep_alive ? 1 : 2)
! 		("httpReadReply: Excess data from \"%s %s\"\n",
  		RequestMethodStr[httpState->orig_request->method],
  		storeUrl(entry));
  	    fwdComplete(httpState->fwd);
