Index: src/store.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/store.c,v
retrieving revision 1.531.2.4
retrieving revision 1.531.2.5
diff -c -r1.531.2.4 -r1.531.2.5
*** src/store.c	2001/02/22 20:53:31	1.531.2.4
--- src/store.c	2001/03/23 20:19:44	1.531.2.5
***************
*** 1033,1039 ****
       * squid_curtime because it results in a negative served_date.
       */
      if (age > squid_curtime - served_date)
! 	if (squid_curtime < age)
  	    served_date = squid_curtime - age;
      entry->expires = reply->expires;
      entry->lastmod = reply->last_modified;
--- 1033,1039 ----
       * squid_curtime because it results in a negative served_date.
       */
      if (age > squid_curtime - served_date)
! 	if (squid_curtime > age)
  	    served_date = squid_curtime - age;
      entry->expires = reply->expires;
      entry->lastmod = reply->last_modified;
