Index: squid/helpers/basic_auth/LDAP/squid_ldap_auth.c
diff -c squid/helpers/basic_auth/LDAP/squid_ldap_auth.c:1.21.2.21 squid/helpers/basic_auth/LDAP/squid_ldap_auth.c:1.21.2.22
*** squid/helpers/basic_auth/LDAP/squid_ldap_auth.c:1.21.2.21	Fri Apr 22 14:29:29 2005
--- squid/helpers/basic_auth/LDAP/squid_ldap_auth.c	Thu Sep  1 16:47:19 2005
***************
*** 672,685 ****
      }
  
      if (debug)
! 	fprintf(stderr, "attempting to bind to user '%s'\n", dn);
      if (!bind_ld && !bind_once)
  	bind_ld = persistent_ld;
      if (!bind_ld)
  	bind_ld = open_ldap_connection(ldapServer, port);
!     if (passwdattr && ldap_compare_s(bind_ld, dn, passwdattr, password) != LDAP_COMPARE_TRUE)
! 	ret = 1;
!     else if (ldap_simple_bind_s(bind_ld, dn, password) != LDAP_SUCCESS)
  	ret = 1;
      if (bind_ld != persistent_ld) {
  	ldap_unbind(bind_ld);
--- 672,687 ----
      }
  
      if (debug)
! 	fprintf(stderr, "attempting to authenticate user '%s'\n", dn);
      if (!bind_ld && !bind_once)
  	bind_ld = persistent_ld;
      if (!bind_ld)
  	bind_ld = open_ldap_connection(ldapServer, port);
!     if (passwdattr) {
! 	if (ldap_compare_s(bind_ld, dn, passwdattr, password) != LDAP_COMPARE_TRUE) {
! 	    ret = 1;
! 	}
!     } else if (ldap_simple_bind_s(bind_ld, dn, password) != LDAP_SUCCESS)
  	ret = 1;
      if (bind_ld != persistent_ld) {
  	ldap_unbind(bind_ld);
