--- squid.conf.default	2004-08-24 11:51:14.723512904 +0200
+++ squid.conf	2004-08-24 11:51:29.986192624 +0200
@@ -688,8 +688,12 @@
 #	Note that for coss, max-size must be less than COSS_MEMBUF_SZ
 #	(hard coded at 1 MB).
 #
+#	NOTE: Remeber to initialize cache directories:
+#	      squid -z -F
+#
 #Default:
 # cache_dir ufs /var/cache/squid 100 16 256
+cache_dir diskd /var/cache/squid 100 16 256
 
 #  TAG: cache_access_log
 #	Logs the client request activity.  Contains an entry for
@@ -804,6 +808,7 @@
 #
 #Default:
 # pid_filename /var/log/squid/squid.pid
+pid_filename /var/run/squid.pid
 
 #  TAG: debug_options
 #	Logging options are set as section,level where each source file
@@ -2025,6 +2030,8 @@
 #
 #Default:
 # cache_effective_user nobody
+cache_effective_user squid
+cache_effective_group squid
 
 #  TAG: visible_hostname
 #	If you want to present a special hostname in error messages, etc,
@@ -2200,8 +2207,14 @@
 #	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 #	<pid>'.
 #
+#	Default value for logfile_rotate is 10, but PLD uses 'logrotate',
+#	which is more configurable, to perform this task. Setting this
+#	value to '0' disables built-in log rotation and allows 'logrotate'
+#	to do it's job.
+#
 #Default:
 # logfile_rotate 10
+logfile_rotate 0
 
 #  TAG: append_domain
 #	Appends local domain name to hostnames without any dots in
