Index: squid3/src/store_swapout.cc
diff -c squid3/src/store_swapout.cc:1.97 squid3/src/store_swapout.cc:1.96
*** squid3/src/store_swapout.cc:1.97	Sat Sep  6 06:59:04 2003
--- squid3/src/store_swapout.cc	Thu Jul 10 22:02:01 2003
***************
*** 68,73 ****
--- 68,74 ----
  
      if (NULL == sio.getRaw()) {
          e->swap_status = SWAPOUT_NONE;
+         cbdataFree(c);
          storeLog(STORE_LOG_SWAPOUTFAIL, e);
          return;
      }
