Index: squid/src/acl.c
diff -c squid/src/acl.c:1.270.2.13 squid/src/acl.c:1.270.2.14
*** squid/src/acl.c:1.270.2.13	Sat Feb  8 08:53:16 2003
--- squid/src/acl.c	Tue May  6 13:54:28 2003
***************
*** 1080,1086 ****
      debug(28, 8) ("Top is %p, Top->data is %s\n", Top,
  	(char *) (Top != NULL ? (Top)->data : "Unavailable"));
  
!     if (user == NULL)
  	return 0;
  
      if (data->flags.required) {
--- 1080,1086 ----
      debug(28, 8) ("Top is %p, Top->data is %s\n", Top,
  	(char *) (Top != NULL ? (Top)->data : "Unavailable"));
  
!     if (user == NULL || strcmp(user, "-") == 0)
  	return 0;
  
      if (data->flags.required) {
