Index: src/main.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/main.c,v
retrieving revision 1.322.2.3
retrieving revision 1.322.2.4
diff -c -r1.322.2.3 -r1.322.2.4
*** src/main.c	2001/01/12 00:51:50	1.322.2.3
--- src/main.c	2001/03/28 23:25:51	1.322.2.4
***************
*** 909,914 ****
--- 909,923 ----
  	if (WIFEXITED(status))
  	    if (WEXITSTATUS(status) == 0)
  		exit(0);
+ 	if (WIFSIGNALED(status)) {
+ 	    switch (WTERMSIG(status)) {
+ 	    case SIGKILL:
+ 		exit(0);
+ 		break;
+ 	    default:
+ 		break;
+ 	    }
+ 	}
  	squid_signal(SIGINT, SIG_DFL, SA_RESTART);
  	sleep(3);
      }
