Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.16 squid/src/cache_cf.c:1.396.2.17
*** squid/src/cache_cf.c:1.396.2.16	Sat Dec  6 10:19:36 2003
--- squid/src/cache_cf.c	Thu Apr 29 17:56:50 2004
***************
*** 502,507 ****
--- 502,511 ----
  	self_destruct();
      if ((token = strtok(NULL, w_space)) == NULL)
  	self_destruct();
+     if (strcmp(token, "none") == 0 || strcmp(token, "-1") == 0) {
+ 	*bptr = (size_t) - 1;
+ 	return;
+     }
      d = atof(token);
      m = u;			/* default to 'units' if none specified */
      if (0.0 == d)
