=== modified file 'src/adaptation/ecap/MessageRep.cc'
--- a/src/adaptation/ecap/MessageRep.cc	2011-04-03 11:05:39 +0000
+++ b/src/adaptation/ecap/MessageRep.cc	2012-06-19 07:37:37 +0000
@@ -150,6 +150,8 @@
         return protocolInternal;
     case PROTO_ICY:
         return Name();
+    case PROTO_COAP:
+    case PROTO_COAPS: // use 'unknown'/none until libecap supports coap:// and coaps://
     case PROTO_NONE:
         return Name();
 

