Index: squid/src/http.c
diff -c squid/src/http.c:1.384.2.15 squid/src/http.c:1.384.2.16
*** squid/src/http.c:1.384.2.15	Sat Apr 24 08:07:08 2004
--- squid/src/http.c	Tue Jun  8 04:54:07 2004
***************
*** 65,71 ****
      if (httpState == NULL)
  	return;
      if (httpState->body_buf) {
! 	clientAbortBody(httpState->orig_request);
  	if (httpState->body_buf) {
  	    memFree(httpState->body_buf, MEM_8K_BUF);
  	    httpState->body_buf = NULL;
--- 65,73 ----
      if (httpState == NULL)
  	return;
      if (httpState->body_buf) {
! 	if (httpState->orig_request->body_connection) {
! 	    clientAbortBody(httpState->orig_request);
! 	}
  	if (httpState->body_buf) {
  	    memFree(httpState->body_buf, MEM_8K_BUF);
  	    httpState->body_buf = NULL;
