Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.14 squid/src/cache_cf.c:1.396.2.15
*** squid/src/cache_cf.c:1.396.2.14	Tue Jul  1 14:42:41 2003
--- squid/src/cache_cf.c	Thu Nov  6 09:50:37 2003
***************
*** 504,509 ****
--- 504,511 ----
      else if ((m = parseBytesUnits(token)) == 0)
  	self_destruct();
      *bptr = m * d / u;
+     if ((double) *bptr != m * d / u)
+ 	self_destruct();
  }
  
  static size_t
