Index: squid/src/peer_digest.c
diff -c squid/src/peer_digest.c:1.83 squid/src/peer_digest.c:1.83.2.1
*** squid/src/peer_digest.c:1.83	Wed Oct 24 00:55:44 2001
--- squid/src/peer_digest.c	Wed Jul 16 14:22:39 2003
***************
*** 688,694 ****
  
      if (fetch->old_entry) {
  	debug(72, 2) ("peerDigestFetchFinish: deleting old entry\n");
! 	storeUnregister(fetch->sc, fetch->old_entry, fetch);
  	storeReleaseRequest(fetch->old_entry);
  	storeUnlockObject(fetch->old_entry);
  	fetch->old_entry = NULL;
--- 688,694 ----
  
      if (fetch->old_entry) {
  	debug(72, 2) ("peerDigestFetchFinish: deleting old entry\n");
! 	storeUnregister(fetch->old_sc, fetch->old_entry, fetch);
  	storeReleaseRequest(fetch->old_entry);
  	storeUnlockObject(fetch->old_entry);
  	fetch->old_entry = NULL;
