Index: squid/src/client_side.c
diff -c squid/src/client_side.c:1.561.2.73 squid/src/client_side.c:1.561.2.74
*** squid/src/client_side.c:1.561.2.73	Mon Mar  7 06:31:35 2005
--- squid/src/client_side.c	Wed Mar  9 07:09:03 2005
***************
*** 1593,1599 ****
  	return;
      }
      http->flags.hit = 1;
!     if (checkNegativeHit(e)) {
  	http->log_type = LOG_TCP_NEGATIVE_HIT;
  	clientSendMoreData(data, buf, size);
      } else if (!Config.onoff.offline && refreshCheckHTTP(e, r) && !http->flags.internal) {
--- 1593,1603 ----
  	return;
      }
      http->flags.hit = 1;
!     if (checkNegativeHit(e)
! #if HTTP_VIOLATIONS
! 	&& !r->flags.nocache_hack
! #endif
! 	) {
  	http->log_type = LOG_TCP_NEGATIVE_HIT;
  	clientSendMoreData(data, buf, size);
      } else if (!Config.onoff.offline && refreshCheckHTTP(e, r) && !http->flags.internal) {
