Index: squid3/src/fs/coss/store_dir_coss.cc
diff -c squid3/src/fs/coss/store_dir_coss.cc:1.50 squid3/src/fs/coss/store_dir_coss.cc:1.51
*** squid3/src/fs/coss/store_dir_coss.cc:1.50	Sun Aug 31 15:20:10 2003
--- squid3/src/fs/coss/store_dir_coss.cc	Sun Aug 31 21:49:42 2003
***************
*** 841,846 ****
--- 841,848 ----
  
      max_size = size;
  
+     parseOptions(0);
+ 
      /* Enforce maxobjsize being set to something */
      if (max_objsize == -1)
          fatal("COSS requires max-size to be set to something other than -1!\n");
