Index: squid/src/acl.c
diff -c squid/src/acl.c:1.270.2.29 squid/src/acl.c:1.270.2.30
*** squid/src/acl.c:1.270.2.29	Sat Sep 25 05:56:16 2004
--- squid/src/acl.c	Wed Oct 20 17:18:36 2004
***************
*** 643,649 ****
      q = xcalloc(1, sizeof(acl_hdr_data));
      q->hdr_name = xstrdup(t);
      q->hdr_id = httpHeaderIdByNameDef(t, strlen(t));
!     aclParseRegexList(q->reglist);
      if (!q->reglist) {
  	debug(28, 0) ("%s line %d: %s\n", cfg_filename, config_lineno, config_input_line);
  	debug(28, 0) ("aclParseHeader: No pattern defined '%s'\n", t);
--- 643,649 ----
      q = xcalloc(1, sizeof(acl_hdr_data));
      q->hdr_name = xstrdup(t);
      q->hdr_id = httpHeaderIdByNameDef(t, strlen(t));
!     aclParseRegexList(&q->reglist);
      if (!q->reglist) {
  	debug(28, 0) ("%s line %d: %s\n", cfg_filename, config_lineno, config_input_line);
  	debug(28, 0) ("aclParseHeader: No pattern defined '%s'\n", t);
***************
*** 693,705 ****
  aclDumpHeader(acl_hdr_data * hd)
  {
      wordlist *W = NULL;
!     relist *data = hd->reglist;
!     wordlistAdd(&W, httpHeaderNameById(hd->hdr_id));
!     while (data != NULL) {
! 	wordlistAdd(&W, data->pattern);
! 	data = data->next;
      }
!     return aclDumpRegexList(hd->reglist);
  }
  
  #if SQUID_SNMP
--- 693,711 ----
  aclDumpHeader(acl_hdr_data * hd)
  {
      wordlist *W = NULL;
!     while (hd) {
! 	MemBuf mb;
! 	relist *data;
! 	memBufDefInit(&mb);
! 	memBufPrintf(&mb, "%s", hd->hdr_name);
! 	for (data = hd->reglist; data; data = data->next) {
! 	    memBufPrintf(&mb, " %s", data->pattern);
! 	}
! 	wordlistAdd(&W, mb.buf);
! 	memBufClean(&mb);
! 	hd = hd->next;
      }
!     return W;
  }
  
  #if SQUID_SNMP
