Index: squid/configure.in
diff -c squid/configure.in:1.251.2.92 squid/configure.in:1.251.2.93
*** squid/configure.in:1.251.2.92	Mon May 16 16:38:57 2005
--- squid/configure.in	Sun Jul 10 18:39:22 2005
***************
*** 786,801 ****
                            XBS5_LP64_OFF64        64 bits (legacy)
                            XBS5_LPBIG_OFFBIG      large pointers and files (legacy)
                            default                The default for your OS],
! [ case "$enableval" in
    yes|no)
  	echo "--with-build-environment expects a build environment string as used by getconf"
  	exit 1
  	;;
-   _*)
- 	buildmodel="$enableval"
- 	;;
    *)
! 	buildmodel="_$enableval"
  	;;
    esac
  ])
--- 786,798 ----
                            XBS5_LP64_OFF64        64 bits (legacy)
                            XBS5_LPBIG_OFFBIG      large pointers and files (legacy)
                            default                The default for your OS],
! [ case "$withval" in
    yes|no)
  	echo "--with-build-environment expects a build environment string as used by getconf"
  	exit 1
  	;;
    *)
! 	buildmodel="$withval"
  	;;
    esac
  ])
***************
*** 813,820 ****
  		CFLAGS="-D_FILE_OFFSET_BITS=64 $CFLAGS"
  	fi
  fi
! if test -n "$buildmodel" && test "$buildmodel" != "_default"; then
  	echo "Using $buildmodel build environment"
  	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
  	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
  	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
--- 810,823 ----
  		CFLAGS="-D_FILE_OFFSET_BITS=64 $CFLAGS"
  	fi
  fi
! if test -n "$buildmodel" && test "$buildmodel" != "default"; then
  	echo "Using $buildmodel build environment"
+ 	if test "`getconf _$buildmodel 2>/dev/null || true`" = 1 || test "`getconf $buildmodel 2>/dev/null || true`" ; then
+ 	    : # All fine
+ 	else
+ 	    echo "ERROR: Build environment $buildmodel not known to getconf."
+ 	    exit 1
+ 	fi
  	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
  	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
  	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
Index: squid/configure
diff -c squid/configure:1.248.2.94 squid/configure:1.248.2.95
*** squid/configure:1.248.2.94	Mon May 16 16:39:29 2005
--- squid/configure	Sun Jul 10 18:40:15 2005
***************
*** 2637,2652 ****
  # Check whether --with-build-environment or --without-build-environment was given.
  if test "${with_build_environment+set}" = set; then
    withval="$with_build_environment"
!    case "$enableval" in
    yes|no)
  	echo "--with-build-environment expects a build environment string as used by getconf"
  	exit 1
  	;;
-   _*)
- 	buildmodel="$enableval"
- 	;;
    *)
! 	buildmodel="_$enableval"
  	;;
    esac
  
--- 2637,2649 ----
  # Check whether --with-build-environment or --without-build-environment was given.
  if test "${with_build_environment+set}" = set; then
    withval="$with_build_environment"
!    case "$withval" in
    yes|no)
  	echo "--with-build-environment expects a build environment string as used by getconf"
  	exit 1
  	;;
    *)
! 	buildmodel="$withval"
  	;;
    esac
  
***************
*** 2666,2673 ****
  		CFLAGS="-D_FILE_OFFSET_BITS=64 $CFLAGS"
  	fi
  fi
! if test -n "$buildmodel" && test "$buildmodel" != "_default"; then
  	echo "Using $buildmodel build environment"
  	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
  	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
  	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
--- 2663,2676 ----
  		CFLAGS="-D_FILE_OFFSET_BITS=64 $CFLAGS"
  	fi
  fi
! if test -n "$buildmodel" && test "$buildmodel" != "default"; then
  	echo "Using $buildmodel build environment"
+ 	if test "`getconf _$buildmodel 2>/dev/null || true`" = 1 || test "`getconf $buildmodel 2>/dev/null || true`" ; then
+ 	    : # All fine
+ 	else
+ 	    echo "ERROR: Build environment $buildmodel not known to getconf."
+ 	    exit 1
+ 	fi
  	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
  	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
  	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
