--- squid-2.3.STABLE4/QUICKSTART.loc	Fri Jan 19 12:47:07 2001
+++ squid-2.3.STABLE4/QUICKSTART	Fri Jan 19 12:48:22 2001
@@ -7,10 +7,9 @@
 INSTALL in the same directory), you have to configure the squid.conf
 file. This is the list of the values you *need* to change, because no
 sensible defaults could be defined. Do not touch the other variables
-for now.  We assume you have installed Squid in the default location:
-/usr/local/squid
+for now.
 
-Uncomment and edit the following lines in /usr/local/squid/etc/squid.conf:
+Uncomment and edit the following lines in /etc/squid/squid.conf:
 
 ==============================================================================
 
@@ -26,7 +25,7 @@
     Warning: Squid uses much more than this value.  Rule of thumb: if
     you have N megabytes free for Squid, put N/3 here.
 
-cache_dir /usr/local/squid/cache 100 16 256
+cache_dir /var/spool/squid 100 16 256
 
     Add here (first number, here 100) the amount of hard disk space 
     (in megabytes) to devote to caching.
@@ -70,10 +69,10 @@
 After editing squid.conf to your liking, run Squid from the command
 line TWICE:
 
-    % /usr/local/squid/bin/squid -z
-    % /usr/local/squid/bin/squid 
+    % /usr/sbin/squid -z
+    % /usr/sbin/squid 
 
-Check in the cache.log (/usr/local/squid/logs/cache.log) that
+Check in the cache.log (/var/log/squid/cache.log) that
 everything is all right.  Note that "WARNING: Cannot write to swap
 directory" is normal the first time you run Squid.
 
