--- ./configure.in.org	2006-12-03 22:02:15.000000000 +0000
+++ ./configure.in	2006-12-03 22:04:56.000000000 +0000
@@ -29,6 +29,7 @@
 
 
 CRYPTLIB=''
+NSLLIB=''
 REGEXLIB=''	# -lregex
 LIBREGEX=''	# libregex.a
 
@@ -2011,8 +2012,10 @@
  fi
 
 dnl Check for needed libraries
-AC_CHECK_LIB(nsl, main)
-AC_CHECK_LIB(socket, main)
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname))
+AC_CHECK_LIB(nsl, yp_match, NSLLIB="-lnsl")
+AC_SUBST(NSLLIB)
+AC_SEARCH_LIBS(socket, socket)
 AC_MSG_CHECKING(for winsock)
 save_LIBS="$LIBS"
 for curlib in ws2_32 wsock32; do
--- ./helpers/basic_auth/YP/Makefile.am.org	2005-05-17 16:56:26.000000000 +0000
+++ ./helpers/basic_auth/YP/Makefile.am	2006-12-03 22:06:20.000000000 +0000
@@ -7,5 +7,5 @@
 
 libexec_PROGRAMS = yp_auth
 yp_auth_SOURCES	= yp_auth.c nis_support.h nis_support.c
-LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(XTRA_LIBS)
+LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(NSLLIB) $(XTRA_LIBS)
 INCLUDES	= -I$(top_srcdir)/include
