Index: squid/configure.in
diff -c squid/configure.in:1.251.2.101 squid/configure.in:1.251.2.102
*** squid/configure.in:1.251.2.101	Thu Sep 22 02:15:05 2005
--- squid/configure.in	Wed Sep 28 14:50:25 2005
***************
*** 68,74 ****
--- 68,76 ----
  
  dnl Substitutions
  AC_DEFINE_UNQUOTED(CACHE_HTTP_PORT, $CACHE_HTTP_PORT)
+ AC_SUBST(CACHE_HTTP_PORT)
  AC_DEFINE_UNQUOTED(CACHE_ICP_PORT, $CACHE_ICP_PORT)
+ AC_SUBST(CACHE_ICP_PORT)
  
  AC_DEFINE_UNQUOTED(CONFIG_HOST_TYPE, "$host")
  
Index: squid/src/Makefile.am
diff -c squid/src/Makefile.am:1.16.2.11 squid/src/Makefile.am:1.16.2.12
*** squid/src/Makefile.am:1.16.2.11	Fri Apr 22 19:32:27 2005
--- squid/src/Makefile.am	Wed Sep 28 14:50:25 2005
***************
*** 272,277 ****
--- 272,279 ----
  	cachemgr.conf \
  	mime.conf.default
  
+ DEFAULT_HTTP_PORT	= @CACHE_HTTP_PORT@
+ DEFAULT_ICP_PORT	= @CACHE_ICP_PORT@
  DEFAULT_PREFIX		= $(prefix)
  DEFAULT_CONFIG_FILE     = $(sysconfdir)/squid.conf
  DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
***************
*** 325,330 ****
--- 327,334 ----
  ## depend on the Makefile. 
  cf.data: cf.data.pre Makefile
  	sed "\
+ 	s%@DEFAULT_HTTP_PORT@%$(DEFAULT_HTTP_PORT)%g;\
+ 	s%@DEFAULT_ICP_PORT@%$(DEFAULT_ICP_PORT)%g;\
  	s%@DEFAULT_MIME_TABLE@%$(DEFAULT_MIME_TABLE)%g;\
  	s%@DEFAULT_DNSSERVER@%$(DEFAULT_DNSSERVER)%g;\
  	s%@DEFAULT_UNLINKD@%$(DEFAULT_UNLINKD)%g;\
Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.100 squid/src/cf.data.pre:1.245.2.101
*** squid/src/cf.data.pre:1.245.2.100	Wed Sep 28 06:29:45 2005
--- squid/src/cf.data.pre	Wed Sep 28 14:50:25 2005
***************
*** 56,62 ****
  NAME: http_port ascii_port
  TYPE: sockaddr_in_list
  DEFAULT: none
! DEFAULT_IF_NONE: 3128
  LOC: Config.Sockaddr.http
  DOC_START
  	Usage:	port
--- 56,62 ----
  NAME: http_port ascii_port
  TYPE: sockaddr_in_list
  DEFAULT: none
! DEFAULT_IF_NONE: @DEFAULT_HTTP_PORT@
  LOC: Config.Sockaddr.http
  DOC_START
  	Usage:	port
***************
*** 145,151 ****
  
  NAME: icp_port udp_port
  TYPE: ushort
! DEFAULT: 3130
  LOC: Config.Port.icp
  DOC_START
  	The port number where Squid sends and receives ICP queries to
--- 145,151 ----
  
  NAME: icp_port udp_port
  TYPE: ushort
! DEFAULT: @DEFAULT_ICP_PORT@
  LOC: Config.Port.icp
  DOC_START
  	The port number where Squid sends and receives ICP queries to
Index: squid/configure
diff -c squid/configure:1.248.2.103 squid/configure:1.248.2.104
*** squid/configure:1.248.2.103	Thu Sep 22 02:15:26 2005
--- squid/configure	Wed Sep 28 14:57:13 2005
***************
*** 9055,9060 ****
--- 9057,9064 ----
  s%@CGIEXT@%$CGIEXT%g
  s%@ENABLE_WIN32SPECIFIC_TRUE@%$ENABLE_WIN32SPECIFIC_TRUE%g
  s%@ENABLE_WIN32SPECIFIC_FALSE@%$ENABLE_WIN32SPECIFIC_FALSE%g
+ s%@CACHE_HTTP_PORT@%$CACHE_HTTP_PORT%g
+ s%@CACHE_ICP_PORT@%$CACHE_ICP_PORT%g
  s%@LIBDLMALLOC@%$LIBDLMALLOC%g
  s%@LIB_MALLOC@%$LIB_MALLOC%g
  s%@STORE_OBJS@%$STORE_OBJS%g
Index: squid/src/Makefile.in
diff -c squid/src/Makefile.in:1.225.2.24 squid/src/Makefile.in:1.225.2.25
*** squid/src/Makefile.in:1.225.2.24	Sat Apr 23 18:12:08 2005
--- squid/src/Makefile.in	Wed Sep 28 14:57:21 2005
***************
*** 73,78 ****
--- 73,80 ----
  AUTH_OBJS = @AUTH_OBJS@
  AWK = @AWK@
  BASIC_AUTH_HELPERS = @BASIC_AUTH_HELPERS@
+ CACHE_HTTP_PORT = @CACHE_HTTP_PORT@
+ CACHE_ICP_PORT = @CACHE_ICP_PORT@
  CC = @CC@
  CGIEXT = @CGIEXT@
  CPP = @CPP@
***************
*** 375,380 ****
--- 377,384 ----
  	mime.conf.default
  
  
+ DEFAULT_HTTP_PORT = @CACHE_HTTP_PORT@
+ DEFAULT_ICP_PORT = @CACHE_ICP_PORT@
  DEFAULT_PREFIX = $(prefix)
  DEFAULT_CONFIG_FILE = $(sysconfdir)/squid.conf
  DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
***************
*** 1144,1149 ****
--- 1148,1155 ----
  
  cf.data: cf.data.pre Makefile
  	sed "\
+ 	s%@DEFAULT_HTTP_PORT@%$(DEFAULT_HTTP_PORT)%g;\
+ 	s%@DEFAULT_ICP_PORT@%$(DEFAULT_ICP_PORT)%g;\
  	s%@DEFAULT_MIME_TABLE@%$(DEFAULT_MIME_TABLE)%g;\
  	s%@DEFAULT_DNSSERVER@%$(DEFAULT_DNSSERVER)%g;\
  	s%@DEFAULT_UNLINKD@%$(DEFAULT_UNLINKD)%g;\
