Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.21 squid/src/cache_cf.c:1.396.2.22
*** squid/src/cache_cf.c:1.396.2.21	Fri Mar  4 09:22:16 2005
--- squid/src/cache_cf.c	Wed Mar  9 07:25:49 2005
***************
*** 517,523 ****
      else if ((m = parseBytesUnits(token)) == 0)
  	self_destruct();
      *bptr = m * d / u;
!     if ((double) *bptr != m * d / u)
  	self_destruct();
  }
  
--- 517,523 ----
      else if ((m = parseBytesUnits(token)) == 0)
  	self_destruct();
      *bptr = m * d / u;
!     if ((double) *bptr * 2 != m * d / u * 2)
  	self_destruct();
  }
  
