--- squid-3.0-PRE3/src/access_log.cc.bak	2004-02-18 16:12:33.000000000 +0100
+++ squid-3.0-PRE3/src/access_log.cc	2004-02-18 16:17:53.000000000 +0100
@@ -1557,7 +1557,7 @@
     if (NULL == hash)
         return;
 
-    fv = hash_lookup(hash, key);
+    fv = (fvdb_entry *)hash_lookup(hash, key);
 
     if (NULL == fv) {
         fv = static_cast <fvdb_entry *>(xcalloc(1, sizeof(fvdb_entry)));
@@ -1613,7 +1613,7 @@
 void
 fvdbFreeEntry(void *data)
 {
-    fvdb_entry *fv = data;
+    fvdb_entry *fv = (fvdb_entry *)data;
     xfree(fv->hash.key);
     xfree(fv);
 }
