diff -Nur squid-2.2.STABLE3/include/version.h squid-2.2.STABLE3.pld/include/version.h
--- squid-2.2.STABLE3/include/version.h	Wed May 19 18:06:54 1999
+++ squid-2.2.STABLE3.pld/include/version.h	Wed Jun  9 13:19:20 1999
@@ -4,7 +4,7 @@
  *  SQUID_VERSION - String for version id of this distribution
  */
 #ifndef SQUID_VERSION
-#define SQUID_VERSION	"2.2.STABLE4"
+#define SQUID_VERSION	"PLD Linux"
 #endif
 
 #ifndef SQUID_RELEASE_TIME
