Index: squid3/src/protos.h
diff -c squid3/src/protos.h:1.490 squid3/src/protos.h:1.491
*** squid3/src/protos.h:1.490	Sun Aug 10 05:00:44 2003
--- squid3/src/protos.h	Sun Aug 17 07:42:04 2003
***************
*** 1041,1047 ****
--- 1041,1050 ----
  
  SQUIDCEXTERN int getrusage(int, struct rusage *);
  SQUIDCEXTERN int getpagesize(void);
+ #if !defined(_XPG4_2) && !(defined(__EXTENSIONS__) || \
+         (!defined(_POSIX_C_SOURCE) && !defined(_XOPEN_SOURCE)))
  SQUIDCEXTERN int gethostname(char *, int);
+ #endif
  #endif
  
  /*
