Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.63 squid/src/cf.data.pre:1.245.2.64
*** squid/src/cf.data.pre:1.245.2.63	Mon Apr  5 11:24:59 2004
--- squid/src/cf.data.pre	Sun Apr 11 14:27:57 2004
***************
*** 1342,1349 ****
  	"program" cmdline
  	Specify the command for the external authenticator.  Such a program
  	reads a line containing "username":"realm" and replies with the
! 	appropriate H(A1) value base64 encoded.  See rfc 2616 for the
! 	definition of H(A1).
  
  	By default, the digest authentication scheme is not used unless a
  	program is specified.
--- 1342,1349 ----
  	"program" cmdline
  	Specify the command for the external authenticator.  Such a program
  	reads a line containing "username":"realm" and replies with the
! 	appropriate H(A1) value base64 encoded or ERR if the user (or his H(A1)
! 	hash) does not exists.  See rfc 2616 for the definition of H(A1).
  
  	By default, the digest authentication scheme is not used unless a
  	program is specified.
