Index: src/acl.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/acl.c,v
retrieving revision 1.205.2.17
retrieving revision 1.205.2.18
diff -c -r1.205.2.17 -r1.205.2.18
*** src/acl.c	2000/03/02 17:29:14	1.205.2.17
--- src/acl.c	2000/08/10 21:36:17	1.205.2.18
***************
*** 391,397 ****
  #define SCAN_ACL1       "%[0123456789.]-%[0123456789.]/%[0123456789.]"
  #define SCAN_ACL2       "%[0123456789.]-%[0123456789.]"
  #define SCAN_ACL3       "%[0123456789.]/%[0123456789.]"
! #define SCAN_ACL4       "%[0123456789.]"
  
  static acl_ip_data *
  aclParseIpData(const char *t)
--- 391,397 ----
  #define SCAN_ACL1       "%[0123456789.]-%[0123456789.]/%[0123456789.]"
  #define SCAN_ACL2       "%[0123456789.]-%[0123456789.]"
  #define SCAN_ACL3       "%[0123456789.]/%[0123456789.]"
! #define SCAN_ACL4       "%[0123456789.]%c"
  
  static acl_ip_data *
  aclParseIpData(const char *t)
***************
*** 404,409 ****
--- 404,410 ----
      acl_ip_data **Q;
      struct hostent *hp;
      char **x;
+     char c;
      debug(28, 5) ("aclParseIpData: %s\n", t);
      if (!strcasecmp(t, "all")) {
  	q->addr1.s_addr = 0;
***************
*** 417,423 ****
  	mask[0] = '\0';
      } else if (sscanf(t, SCAN_ACL3, addr1, mask) == 2) {
  	addr2[0] = '\0';
!     } else if (sscanf(t, SCAN_ACL4, addr1) == 1) {
  	addr2[0] = '\0';
  	mask[0] = '\0';
      } else if (sscanf(t, "%[^/]/%s", addr1, mask) == 2) {
--- 418,424 ----
  	mask[0] = '\0';
      } else if (sscanf(t, SCAN_ACL3, addr1, mask) == 2) {
  	addr2[0] = '\0';
!     } else if (sscanf(t, SCAN_ACL4, addr1, &c) == 1) {
  	addr2[0] = '\0';
  	mask[0] = '\0';
      } else if (sscanf(t, "%[^/]/%s", addr1, mask) == 2) {
