Index: squid3/src/SwapDir.cc
diff -c squid3/src/SwapDir.cc:1.4 squid3/src/SwapDir.cc:1.5
*** squid3/src/SwapDir.cc:1.4	Tue Jul 22 09:23:01 2003
--- squid3/src/SwapDir.cc	Sun Aug 31 15:20:08 2003
***************
*** 154,159 ****
--- 154,161 ----
          if (value)
              *value++ = '\0';	/* cut on = */
  
+         debugs(3,2, "SwapDir::parseOptions: parsing store option '" << name << "'='" << (value ? value : "") << "'");
+ 
          if (newOption)
              if (!newOption->parse(name, value, reconfiguring))
                  self_destruct();
Index: squid3/src/fs/coss/store_dir_coss.cc
diff -c squid3/src/fs/coss/store_dir_coss.cc:1.49 squid3/src/fs/coss/store_dir_coss.cc:1.50
*** squid3/src/fs/coss/store_dir_coss.cc:1.49	Sat Aug 30 00:39:24 2003
--- squid3/src/fs/coss/store_dir_coss.cc	Sun Aug 31 15:20:10 2003
***************
*** 49,57 ****
  
  typedef struct _RebuildState RebuildState;
  
- void storeCossDirParseBlkSize(SwapDir *, const char *, const char *, int);
- void storeCossDirDumpBlkSize(StoreEntry *, const char *, const SwapDir *);
- 
  struct _RebuildState
  {
      CossSwapDir *sd;
--- 49,54 ----
***************
*** 900,906 ****
  }
  
  
! CossSwapDir::CossSwapDir() : SwapDir ("coss"), fd (-1), swaplog_fd(-1), count(0), current_membuf (NULL), current_offset(0), numcollisions(0)
  {
      membufs.head = NULL;
      membufs.tail = NULL;
--- 897,903 ----
  }
  
  
! CossSwapDir::CossSwapDir() : SwapDir ("coss"), fd (-1), swaplog_fd(-1), count(0), current_membuf (NULL), current_offset(0), numcollisions(0), blksz_bits(0)
  {
      membufs.head = NULL;
      membufs.tail = NULL;
***************
*** 912,917 ****
--- 909,922 ----
  bool
  CossSwapDir::optionBlockSizeParse(const char *option, const char *value, int reconfiguring)
  {
+     assert(option);
+ 
+     if (strcmp(option, "block-size") != 0)
+         return false;
+ 
+     if (!value)
+         self_destruct();
+ 
      int blksz = atoi(value);
  
      if (blksz == (1 << blksz_bits))
