Index: src/dns_internal.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/dns_internal.c,v
retrieving revision 1.13.2.13
retrieving revision 1.13.2.14
diff -c -r1.13.2.13 -r1.13.2.14
*** src/dns_internal.c	2000/07/18 03:48:07	1.13.2.13
--- src/dns_internal.c	2000/07/22 17:57:24	1.13.2.14
***************
*** 197,202 ****
--- 197,203 ----
      storeAppendPrintf(sentry, "RCODE");
      for (i = 0; i < MAX_ATTEMPT; i++)
  	storeAppendPrintf(sentry, " ATTEMPT%d", i + 1);
+     storeAppendPrintf(sentry, "\n");
      for (j = 0; j < MAX_RCODE; j++) {
  	storeAppendPrintf(sentry, "%5d", j);
  	for (i = 0; i < MAX_ATTEMPT; i++)
