--- squid-2.4.STABLE2/configure.in~	Fri Oct 12 18:34:14 2001
+++ squid-2.4.STABLE2/configure.in	Fri Oct 12 19:17:42 2001
@@ -16,6 +16,7 @@
 AC_CANONICAL_HOST
 
 CRYPTLIB=''
+NSLLIB=''
 REGEXLIB=''	# -lregex
 LIBREGEX=''	# libregex.a
 
@@ -1006,8 +1007,10 @@
  fi
 
 dnl Check for needed libraries
-AC_CHECK_LIB(nsl, main)
 AC_CHECK_LIB(socket, main)
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname)) 
+AC_CHECK_LIB(nsl, yp_match, NSLLIB="-lnsl")
+AC_SUBST(NSLLIB)
 
 if test "x$ac_cv_enabled_dlmalloc" = "xyes" ; then
   echo "skipping libmalloc check (--enable-dlmalloc specified)"
--- squid-2.4.STABLE2/auth_modules/YP/Makefile.in~	Tue Jan 16 22:12:28 2001
+++ squid-2.4.STABLE2/auth_modules/YP/Makefile.in	Fri Oct 12 19:35:32 2001
@@ -35,6 +35,7 @@
 PERL            = @PERL@
 CRYPTLIB	= @CRYPTLIB@
 REGEXLIB	= @REGEXLIB@
+NSLLIB		= @NSLLIB@
 PTHREADLIB	= @PTHREADLIB@
 SNMPLIB		= @SNMPLIB@
 MALLOCLIB	= @LIB_MALLOC@
@@ -49,7 +49,7 @@
 
 INCLUDE		= -I. -I../../include -I$(top_srcdir)/include
 CFLAGS 		= $(AC_CFLAGS) $(INCLUDE) $(DEFINES)
-AUTH_LIBS	= -L../../lib -lmiscutil $(CRYPTLIB) $(XTRA_LIBS)
+AUTH_LIBS	= -L../../lib -lmiscutil $(CRYPTLIB) $(NSLLIB) $(XTRA_LIBS)
 
 PROGS           = $(YP_AUTH_EXE)
 OBJS	 	= yp_auth.o nis_support.o
