Index: squid/src/acl.c
diff -c squid/src/acl.c:1.270.2.10 squid/src/acl.c:1.270.2.12
*** squid/src/acl.c:1.270.2.10	Mon Oct  7 21:33:52 2002
--- squid/src/acl.c	Sat Nov  9 20:57:03 2002
***************
*** 1745,1750 ****
--- 1745,1751 ----
  	 */
  	if (!cbdataValid(A)) {
  	    cbdataUnlock(A);
+ 	    checklist->access_list = NULL;
  	    break;
  	}
  	debug(28, 3) ("aclCheck: checking '%s'\n", A->cfgline);
***************
*** 1818,1826 ****
  	 * is allowed, denied, requires authentication, or we move on to
  	 * the next entry.
  	 */
- 	cbdataUnlock(A);
  	if (match) {
  	    debug(28, 3) ("aclCheck: match found, returning %d\n", allow);
  	    aclCheckCallback(checklist, allow);
  	    return;
  	}
--- 1819,1828 ----
  	 * is allowed, denied, requires authentication, or we move on to
  	 * the next entry.
  	 */
  	if (match) {
  	    debug(28, 3) ("aclCheck: match found, returning %d\n", allow);
+ 	    cbdataUnlock(A);
+ 	    checklist->access_list = NULL;
  	    aclCheckCallback(checklist, allow);
  	    return;
  	}
***************
*** 1830,1835 ****
--- 1832,1838 ----
  	 */
  	if (A->next)
  	    cbdataLock(A->next);
+ 	cbdataUnlock(A);
      }
      debug(28, 3) ("aclCheck: NO match found, returning %d\n", allow != ACCESS_DENIED ? ACCESS_DENIED : ACCESS_ALLOWED);
      aclCheckCallback(checklist, allow != ACCESS_DENIED ? ACCESS_DENIED : ACCESS_ALLOWED);
***************
*** 1844,1849 ****
--- 1847,1856 ----
      if (checklist->conn) {
  	cbdataUnlock(checklist->conn);
  	checklist->conn = NULL;
+     }
+     if (checklist->access_list) {
+ 	cbdataUnlock(checklist->access_list);
+ 	checklist->access_list = NULL;
      }
      aclCheckCleanup(checklist);
      cbdataFree(checklist);
