--- squid-2.5.STABLE7/configure.in.orig	2004-10-22 21:15:07.080978000 +0200
+++ squid-2.5.STABLE7/configure.in	2004-10-22 22:07:51.424648102 +0200
@@ -1520,15 +1520,15 @@
 
 AC_CHECK_LIB(bsd, main)
 AC_CHECK_LIB(regex, main, [REGEXLIB="-lregex"])
-AC_CHECK_LIB(bind, gethostbyname)
-if test $ac_cv_lib_bind_gethostbyname = "no" ; then
+AC_CHECK_FUNC(gethostbyname,,[AC_CHECK_LIB(bind, gethostbyname)])
+if test "$ac_cv_lib_bind_gethostbyname" != "yes" ; then
     case "$host" in
 	i386-*-freebsd*)
 		echo "skipping libresolv checks for $host"
 		;;
 	*)
-		AC_CHECK_LIB(resolv, inet_aton, AC_CHECK_LIB(44bsd, inet_aton))
-		AC_CHECK_LIB(resolv, main)
+		AC_CHECK_FUNC(inet_aton, [], [AC_CHECK_LIB(resolv, inet_aton, AC_CHECK_LIB(44bsd, inet_aton))])
+		dnl AC_CHECK_LIB(resolv, main)
 		;;
     esac
 fi
