Index: squid/src/fs/aufs/aiops.c
diff -c squid/src/fs/aufs/aiops.c:1.12.2.11 squid/src/fs/aufs/aiops.c:1.12.2.12
*** squid/src/fs/aufs/aiops.c:1.12.2.11	Mon Apr 25 10:27:02 2005
--- squid/src/fs/aufs/aiops.c	Mon Dec 26 09:32:28 2005
***************
*** 941,946 ****
--- 941,949 ----
      squidaio_thread_t *threadp;
      int i;
  
+     if (!squidaio_initialised)
+ 	return;
+ 
      storeAppendPrintf(sentry, "\n\nThreads Status:\n");
      storeAppendPrintf(sentry, "#\tID\t# Requests\n");
  
