Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.10 squid/src/cache_cf.c:1.396.2.11
*** squid/src/cache_cf.c:1.396.2.10	Wed Oct 16 20:14:34 2002
--- squid/src/cache_cf.c	Thu Nov 14 13:14:17 2002
***************
*** 2438,2447 ****
  	fromFile = 0;
  	goto strtok_again;
      } else {
  	t = buf;
  	/* skip leading and trailing white space */
  	t += strspn(buf, w_space);
! 	t[strcspn(t, w_space)] = '\0';
  	/* skip comments */
  	if (*t == '#')
  	    goto strtok_again;
--- 2438,2454 ----
  	fromFile = 0;
  	goto strtok_again;
      } else {
+ 	char *t2, *t3;
  	t = buf;
  	/* skip leading and trailing white space */
  	t += strspn(buf, w_space);
! 	t2 = t + strcspn(t, w_space);
! 	t3 = t2 + strspn(t2, w_space);
! 	while (*t3 && *t3 != '#') {
! 	    t2 = t3 + strcspn(t3, w_space);
! 	    t3 = t2 + strspn(t2, w_space);
! 	}
! 	*t2 = '\0';
  	/* skip comments */
  	if (*t == '#')
  	    goto strtok_again;
***************
*** 2451,2454 ****
  	return t;
      }
  }
- 
--- 2458,2460 ----
