Index: squid/src/cache_cf.c
diff -c squid/src/cache_cf.c:1.396.2.28 squid/src/cache_cf.c:1.396.2.29
*** squid/src/cache_cf.c:1.396.2.28	Sun Jun 19 03:36:33 2005
--- squid/src/cache_cf.c	Thu Sep  1 16:04:51 2005
***************
*** 2545,2551 ****
      struct stat sb;
      char pathbuf[BUFSIZ];
      assert(path != NULL);
!     if (Config.chroot_dir) {
  	snprintf(pathbuf, BUFSIZ, "%s/%s", Config.chroot_dir, path);
  	path = pathbuf;
      }
--- 2545,2551 ----
      struct stat sb;
      char pathbuf[BUFSIZ];
      assert(path != NULL);
!     if (Config.chroot_dir && (geteuid() == 0)) {
  	snprintf(pathbuf, BUFSIZ, "%s/%s", Config.chroot_dir, path);
  	path = pathbuf;
      }
