Index: squid/src/ftp.c
diff -c squid/src/ftp.c:1.316.2.10 squid/src/ftp.c:1.316.2.11
*** squid/src/ftp.c:1.316.2.10	Wed Jul 16 07:42:51 2003
--- squid/src/ftp.c	Sun Dec 14 05:12:07 2003
***************
*** 2264,2269 ****
--- 2264,2271 ----
  ftpDataWriteCallback(int fd, char *buf, size_t size, int err, void *data)
  {
      FtpStateData *ftpState = (FtpStateData *) data;
+     if (err == COMM_ERR_CLOSING)
+ 	return;
      if (!err) {
  	/* Shedule the rest of the request */
  	clientReadBody(ftpState->request, ftpState->data.buf, ftpState->data.size, ftpRequestBody, ftpState);
