diff -urN squid-2.5.STABLE1.org/src/ssl_support.c squid-2.5.STABLE1/src/ssl_support.c
--- squid-2.5.STABLE1.org/src/ssl_support.c	Thu Mar 13 19:11:32 2003
+++ squid-2.5.STABLE1/src/ssl_support.c	Thu Mar 13 19:12:12 2003
@@ -137,7 +137,7 @@
 	"NETSCAPE_CA_DN_BUG", SSL_OP_NETSCAPE_CA_DN_BUG
     },
     {
-	"NON_EXPORT_FIRST", SSL_OP_NON_EXPORT_FIRST
+	"NON_EXPORT_FIRST", SSL_OP_CIPHER_SERVER_PREFERENCE
     },
     {
 	"NETSCAPE_DEMO_CIPHER_CHANGE_BUG", SSL_OP_NETSCAPE_DEMO_CIPHER_CHANGE_BUG
