--- squid.conf.default	Thu Mar 22 14:55:47 2001
+++ squid.conf	Thu Mar 22 15:09:11 2001
@@ -595,8 +595,12 @@
 #	the cache_dir lines with the smallest max-size value first and the
 #	ones with no max-size specification last.
 #
+#	NOTE: Remeber to initialize cache directories:
+#	      squid -z -F
+#
 #Default:
 # cache_dir ufs /var/cache/squid 100 16 256
+cache_dir diskd /var/cache/squid 100 16 256
 
 #  TAG: cache_access_log
 #	Logs the client request activity.  Contains an entry for
@@ -1533,6 +1537,8 @@
 #Default:
 # cache_effective_user nobody
 # cache_effective_group nogroup
+cache_effective_user squid
+cache_effective_group squid
 
 #  TAG: visible_hostname
 #	If you want to present a special hostname in error messages, etc,
@@ -1702,8 +1708,14 @@
 #	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 #	<pid>'.
 #
+#	Default value for logfile_rotate is 10, but PLD uses 'logrotate',
+#	which is more configurable, to perform this task. Setting this
+#	value to '0' disables built-in log rotation and allows 'logrotate'
+#	to do it's job.
+#
 #Default:
 # logfile_rotate 10
+logfile_rotate 0
 
 #  TAG: append_domain
 #	Appends local domain name to hostnames without any dots in
