Index: squid/src/mime.c
diff -c squid/src/mime.c:1.102 squid/src/mime.c:1.102.2.1
*** squid/src/mime.c:1.102	Tue Nov 13 10:32:15 2001
--- squid/src/mime.c	Tue Oct  8 10:20:43 2002
***************
*** 402,407 ****
--- 402,408 ----
      const char *type = mimeGetContentType(icon);
      HttpReply *reply;
      http_version_t version;
+     request_t *r;
      if (type == NULL)
  	fatal("Unknown icon format while reading mime.conf\n");
      buf = internalLocalUri("/squid-internal-static/icons/", icon);
***************
*** 429,435 ****
      EBIT_SET(e->flags, ENTRY_SPECIAL);
      storeSetPublicKey(e);
      storeBuffer(e);
!     e->mem_obj->request = requestLink(urlParse(METHOD_GET, url));
      httpReplyReset(reply = e->mem_obj->reply);
      httpBuildVersion(&version, 1, 0);
      httpReplySetHeaders(reply, version, HTTP_OK, NULL,
--- 430,439 ----
      EBIT_SET(e->flags, ENTRY_SPECIAL);
      storeSetPublicKey(e);
      storeBuffer(e);
!     r = urlParse(METHOD_GET, url);
!     if (NULL == r)
! 	fatal("mimeLoadIcon: cannot parse internal URL");
!     e->mem_obj->request = requestLink(r);
      httpReplyReset(reply = e->mem_obj->reply);
      httpBuildVersion(&version, 1, 0);
      httpReplySetHeaders(reply, version, HTTP_OK, NULL,
