diff -ruN squid-2.0.RELEASE-org/scripts/Makefile.in squid-2.0.RELEASE/scripts/Makefile.in
--- squid-2.0.RELEASE-org/scripts/Makefile.in	Wed May 14 23:08:14 1997
+++ squid-2.0.RELEASE/scripts/Makefile.in	Tue Oct  6 18:28:45 1998
@@ -23,11 +23,11 @@
 install:	
 	@if test ! -d $(exec_prefix); then \
 		echo "mkdir $(exec_prefix)"; \
-		mkdir $(exec_prefix); \
+		mkdir -p $(exec_prefix); \
 	fi
 	@if test ! -d $(bindir); then \
 		echo "mkdir $(bindir)"; \
-		mkdir $(bindir); \
+		mkdir -p $(bindir); \
 	fi
 	$(INSTALL_BIN) RunCache $(bindir)
 	$(INSTALL_BIN) RunAccel $(bindir)
diff -ruN squid-2.0.RELEASE-org/src/Makefile.in squid-2.0.RELEASE/src/Makefile.in
--- squid-2.0.RELEASE-org/src/Makefile.in	Tue Aug 18 01:27:15 1998
+++ squid-2.0.RELEASE/src/Makefile.in	Tue Oct  6 18:34:21 1998
@@ -31,10 +31,10 @@
 DEFAULT_CONFIG_FILE     = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE	= $(sysconfdir)/mime.conf
 DEFAULT_DNSSERVER       = $(libexecdir)/$(DNSSERVER_EXE)
-DEFAULT_CACHE_LOG       = $(localstatedir)/logs/cache.log
-DEFAULT_ACCESS_LOG      = $(localstatedir)/logs/access.log
-DEFAULT_STORE_LOG       = $(localstatedir)/logs/store.log
-DEFAULT_PID_FILE        = $(localstatedir)/logs/squid.pid
+DEFAULT_CACHE_LOG       = $(localstatedir)/log/squid/cache.log
+DEFAULT_ACCESS_LOG      = $(localstatedir)/log/squid/access.log
+DEFAULT_STORE_LOG       = $(localstatedir)/log/squid/store.log
+DEFAULT_PID_FILE        = $(localstatedir)/run/squid.pid
 DEFAULT_SWAP_DIR        = $(localstatedir)/cache
 DEFAULT_PINGER		= $(libexecdir)/$(PINGER_EXE)
 DEFAULT_UNLINKD		= $(libexecdir)/$(UNLINKD_EXE)
@@ -258,32 +258,37 @@
 install-mkdirs:
 	-@if test ! -d $(prefix); then \
 		echo "mkdir $(prefix)"; \
-		mkdir $(prefix); \
+		mkdir -p $(prefix); \
 	fi
 	-@if test ! -d $(exec_prefix); then \
 		echo "mkdir $(exec_prefix)"; \
-		mkdir $(exec_prefix); \
+		mkdir -p $(exec_prefix); \
 	fi
 	-@if test ! -d $(bindir); then \
 		echo "mkdir $(bindir)"; \
-		mkdir $(bindir); \
+		mkdir -p $(bindir); \
 	fi
 	-@if test ! -d $(libexecdir); then \
 		echo "mkdir $(libexecdir)"; \
-		mkdir $(libexecdir); \
+		mkdir -p $(libexecdir); \
 	fi
 	-@if test ! -d $(sysconfdir); then \
 		echo "mkdir $(sysconfdir)"; \
-		mkdir $(sysconfdir); \
+		mkdir -p $(sysconfdir); \
 	fi
-	-@if test ! -d $(localstatedir); then \
-		echo "mkdir $(localstatedir)"; \
-		mkdir $(localstatedir); \
-	fi
-	-@if test ! -d $(localstatedir)/logs; then \
-		echo "mkdir $(localstatedir)/logs"; \
-		mkdir $(localstatedir)/logs; \
+	-@if test ! -d $(localstatedir)/log/squid; then \
+		echo "mkdir $(localstatedir)/log/squid"; \
+		mkdir -p $(localstatedir)/log/squid; \
 	fi
+	-@if test ! -d $(localstatedir)/spool/squid; then \
+		echo "mkdir $(localstatedir)/spool/squid"; \
+		mkdir -p $(localstatedir)/spool/squid; \
+	fi
+	-@if test ! -d $(localstatedir)/run; then \
+		echo "mkdir $(localstatedir)/run"; \
+		mkdir -p $(localstatedir)/run; \
+	fi
+	
 
 # Michael Lupp <mike@nemesis.saar.de> wants to know about additions
 # to the install target.
