diff -urN squid-2.5.STABLE2.org/src/Makefile.am squid-2.5.STABLE2/src/Makefile.am
--- squid-2.5.STABLE2.org/src/Makefile.am	Sat Apr 12 12:06:45 2003
+++ squid-2.5.STABLE2/src/Makefile.am	Sat Apr 12 12:16:25 2003
@@ -274,12 +274,12 @@
 DEFAULT_CONFIG_FILE     = $(sysconfdir)/squid.conf
 DEFAULT_MIME_TABLE	= $(sysconfdir)/mime.conf
 DEFAULT_DNSSERVER       = $(libexecdir)/dnsserver$(EXEEXT)
-DEFAULT_LOG_PREFIX	= $(localstatedir)/logs
+DEFAULT_LOG_PREFIX	= $(localstatedir)/log/squid
 DEFAULT_CACHE_LOG       = $(DEFAULT_LOG_PREFIX)/cache.log
 DEFAULT_ACCESS_LOG      = $(DEFAULT_LOG_PREFIX)/access.log
 DEFAULT_STORE_LOG       = $(DEFAULT_LOG_PREFIX)/store.log
 DEFAULT_PID_FILE        = $(DEFAULT_LOG_PREFIX)/squid.pid
-DEFAULT_SWAP_DIR        = $(localstatedir)/cache
+DEFAULT_SWAP_DIR        = $(localstatedir)/cache/squid
 DEFAULT_PINGER		= $(libexecdir)/pinger$(EXEEXT)
 DEFAULT_UNLINKD		= $(libexecdir)/unlinkd$(EXEEXT)
 DEFAULT_DISKD		= $(libexecdir)/diskd$(EXEEXT)
--- squid-2.5.STABLE6/src/Makefile.am.orig	2004-08-25 19:46:13.000000000 +0200
+++ squid-2.5.STABLE6/src/Makefile.am	2004-08-25 19:44:11.000000000 +0200
@@ -278,7 +278,7 @@
 DEFAULT_CACHE_LOG       = $(DEFAULT_LOG_PREFIX)/cache.log
 DEFAULT_ACCESS_LOG      = $(DEFAULT_LOG_PREFIX)/access.log
 DEFAULT_STORE_LOG       = $(DEFAULT_LOG_PREFIX)/store.log
-DEFAULT_PID_FILE        = $(DEFAULT_LOG_PREFIX)/squid.pid
+DEFAULT_PID_FILE        = /var/run/squid.pid
 DEFAULT_SWAP_DIR        = $(localstatedir)/cache/squid
 DEFAULT_PINGER		= $(libexecdir)/pinger$(EXEEXT)
 DEFAULT_UNLINKD		= $(libexecdir)/unlinkd$(EXEEXT)
