Index: src/store_dir.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/store_dir.c,v
retrieving revision 1.115.2.13
retrieving revision 1.115.2.15
diff -c -r1.115.2.13 -r1.115.2.15
*** src/store_dir.c	2001/03/13 19:11:52	1.115.2.13
--- src/store_dir.c	2001/03/28 16:34:41	1.115.2.15
***************
*** 477,482 ****
--- 477,483 ----
      struct statvfs sfs;
      if (statvfs(path, &sfs)) {
  	debug(50, 1) ("%s: %s\n", path, xstrerror());
+ 	*blksize = 2048;
  	return 1;
      }
      *blksize = (int) sfs.f_frsize;
***************
*** 484,493 ****
--- 485,500 ----
      struct statfs sfs;
      if (statfs(path, &sfs)) {
  	debug(50, 1) ("%s: %s\n", path, xstrerror());
+ 	*blksize = 2048;
  	return 1;
      }
      *blksize = (int) sfs.f_bsize;
  #endif
+     /*
+      * Sanity check; make sure we have a meaningful value.
+      */
+     if (*blksize < 512)
+ 	*blksize = 2048;
      return 0;
  }
  
