Index: squid3/src/forward.cc
diff -c squid3/src/forward.cc:1.114 squid3/src/forward.cc:1.115
*** squid3/src/forward.cc:1.114	Sat Sep 20 18:30:47 2003
--- squid3/src/forward.cc	Sun Nov  9 04:05:24 2003
***************
*** 576,585 ****
          ctimeout = fs->_peer->connect_timeout > 0 ? fs->_peer->connect_timeout
                     : Config.Timeout.peer_connect;
  
!         if (fs->_peer->options.originserver) {
              domain = fwdState->request->host;
-             port = fwdState->request->port;
-         }
      } else {
          host = fwdState->request->host;
          port = fwdState->request->port;
--- 576,583 ----
          ctimeout = fs->_peer->connect_timeout > 0 ? fs->_peer->connect_timeout
                     : Config.Timeout.peer_connect;
  
!         if (fs->_peer->options.originserver)
              domain = fwdState->request->host;
      } else {
          host = fwdState->request->host;
          port = fwdState->request->port;
