diff -urN squid-2.7.STABLE4.orig/configure.in squid-2.7.STABLE4/configure.in
--- squid-2.7.STABLE4.orig/configure.in	2008-08-08 21:21:03.000000000 +0200
+++ squid-2.7.STABLE4/configure.in	2008-09-18 12:42:51.646184400 +0200
@@ -29,6 +29,7 @@
 
 
 CRYPTLIB=''
+NSLLIB=''
 REGEXLIB=''	# -lregex
 LIBREGEX=''	# libregex.a
 
@@ -2043,8 +2044,10 @@
  fi
 
 dnl Check for needed libraries
-AC_CHECK_LIB(nsl, main)
-AC_CHECK_LIB(socket, main)
+AC_CHECK_FUNC(gethostbyname, [], AC_CHECK_LIB(nsl, gethostbyname))
+AC_CHECK_LIB(nsl, yp_match, NSLLIB="-lnsl")
+AC_SUBST(NSLLIB)
+AC_SEARCH_LIBS(socket, socket)
 dnl Check for Winsock only on MinGW, on Cygwin we must use emulated BSD socket API
 	case "$host_os" in
 	mingw|mingw32)
diff -urN squid-2.7.STABLE4.orig/helpers/basic_auth/YP/Makefile.am squid-2.7.STABLE4/helpers/basic_auth/YP/Makefile.am
--- squid-2.7.STABLE4.orig/helpers/basic_auth/YP/Makefile.am	2005-05-17 18:56:26.000000000 +0200
+++ squid-2.7.STABLE4/helpers/basic_auth/YP/Makefile.am	2008-09-18 12:44:11.039717339 +0200
@@ -7,5 +7,5 @@
 
 libexec_PROGRAMS = yp_auth
 yp_auth_SOURCES	= yp_auth.c nis_support.h nis_support.c
-LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(XTRA_LIBS)
+LDADD	= -L$(top_builddir)/lib -lmiscutil $(CRYPTLIB) $(NSLLIB) $(XTRA_LIBS)
 INCLUDES	= -I$(top_srcdir)/include
