Index: squid/src/external_acl.c
diff -c squid/src/external_acl.c:1.1.2.19 squid/src/external_acl.c:1.1.2.20
*** squid/src/external_acl.c:1.1.2.19	Thu Feb 27 01:18:04 2003
--- squid/src/external_acl.c	Tue Mar 18 15:05:17 2003
***************
*** 431,437 ****
  	}
  	ch->auth_user_request = NULL;
      }
!     if (!entry) {
  	debug(82, 2) ("aclMatchExternal: %s(\"%s\") = lookup needed\n", acl->def->name, key);
  	ch->state[ACL_EXTERNAL] = ACL_LOOKUP_NEEDED;
  	return 0;
--- 431,437 ----
  	}
  	ch->auth_user_request = NULL;
      }
!     if (!entry || entry->result == -1) {
  	debug(82, 2) ("aclMatchExternal: %s(\"%s\") = lookup needed\n", acl->def->name, key);
  	ch->state[ACL_EXTERNAL] = ACL_LOOKUP_NEEDED;
  	return 0;
