Index: squid3/src/fs/diskd/store_io_diskd.cc
diff -c squid3/src/fs/diskd/store_io_diskd.cc:1.37 squid3/src/fs/diskd/store_io_diskd.cc:1.38
*** squid3/src/fs/diskd/store_io_diskd.cc:1.37	Sat Aug  9 06:31:41 2003
--- squid3/src/fs/diskd/store_io_diskd.cc	Sat Aug 23 06:33:04 2003
***************
*** 630,635 ****
--- 630,636 ----
  
      if (M->newstyle) {
          DiskdFile *theFile = (DiskdFile *)M->callback_data;
+         theFile->RefCountDereference();
          theFile->completed (M);
      } else
          switch (M->mtype) {
***************
*** 668,673 ****
--- 669,675 ----
      static int seq_no = 0;
      M.mtype = mtype;
      M.callback_data = cbdataReference(theFile);
+     theFile->RefCountReference();
      M.size = size;
      M.offset = offset;
      M.status = -1;
