--- squid.conf.default	2010-09-08 13:45:20.000000000 +0200
+++ squid.conf	2010-09-08 17:29:24.765819003 +0200
@@ -43,7 +43,7 @@
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
 # one who can access services on "localhost" is a local user
-#http_access deny to_localhost
+http_access deny to_localhost
 
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
@@ -65,7 +65,8 @@
 hierarchy_stoplist cgi-bin ?
 
 # Uncomment and adjust the following to add a disk cache directory.
-#cache_dir ufs /var/cache/squid 100 16 256
+# NOTE: Remeber to initialize cache directories: squid -z -F
+cache_dir diskd /var/cache/squid 100 16 256
 
 # Leave coredumps in the first cache dir
 coredump_dir /var/cache/squid
@@ -75,3 +76,12 @@
 refresh_pattern ^gopher:	1440	0%	1440
 refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
 refresh_pattern .		0	20%	4320
+
+# Default value for logfile_rotate is 10, but PLD uses 'logrotate',
+# which is more configurable, to perform this task. Setting this
+# value to '0' disables built-in log rotation and allows 'logrotate'
+# to do it's job.
+logfile_rotate 0
+
+cache_effective_user squid
+cache_effective_group squid
