*** squid-2.5.STABLE5/configure.in	Fri Mar 19 02:14:24 2004
--- squid-2.5/configure.in	Fri Mar 19 02:14:39 2004
***************
*** 9,15 ****
  dnl
  AC_INIT(src/main.c)
  AC_CONFIG_AUX_DIR(cfgaux)
! AM_INIT_AUTOMAKE(squid, 2.5.STABLE5-CVS)
  AM_CONFIG_HEADER(include/autoconf.h)
  AC_REVISION($Revision$)dnl
  AC_PREFIX_DEFAULT(/usr/local/squid)
--- 9,15 ----
  dnl
  AC_INIT(src/main.c)
  AC_CONFIG_AUX_DIR(cfgaux)
! AM_INIT_AUTOMAKE(squid, 2.5.STABLE5)
  AM_CONFIG_HEADER(include/autoconf.h)
  AC_REVISION($Revision$)dnl
  AC_PREFIX_DEFAULT(/usr/local/squid)
