Index: src/carp.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/carp.c,v
retrieving revision 1.6.8.4
retrieving revision 1.6.8.5
diff -c -r1.6.8.4 -r1.6.8.5
*** src/carp.c	2000/05/17 23:49:51	1.6.8.4
--- src/carp.c	2000/08/01 21:59:44	1.6.8.5
***************
*** 60,66 ****
       * sum of carp-load-factor's for all cache_peer's in squid.conf
       * must equal 1.0
       */
!     assert(a == 1.0);
      k = 1;
      P_last = 0;
      p = Config.peers;
--- 60,66 ----
       * sum of carp-load-factor's for all cache_peer's in squid.conf
       * must equal 1.0
       */
!     assert(1000 == (int) (1000.0 * a));
      k = 1;
      P_last = 0;
      p = Config.peers;
