Index: squid3/src/mem.cc
diff -c squid3/src/mem.cc:1.82 squid3/src/mem.cc:1.83
*** squid3/src/mem.cc:1.82	Tue Jul 15 05:33:22 2003
--- squid3/src/mem.cc	Sun Sep 14 16:41:24 2003
***************
*** 337,348 ****
      eventAdd("memPoolCleanIdlePools", CleanIdlePools, NULL, clean_interval, 1);
  }
  
! static int mem_idle_limit = 0;
  
  void
  memConfigure(void)
  {
!     int new_pool_limit;
      /* set to configured value first */
  
      if (!Config.onoff.mem_pools)
--- 337,348 ----
      eventAdd("memPoolCleanIdlePools", CleanIdlePools, NULL, clean_interval, 1);
  }
  
! static unsigned int mem_idle_limit = 0;
  
  void
  memConfigure(void)
  {
!     unsigned int new_pool_limit;
      /* set to configured value first */
  
      if (!Config.onoff.mem_pools)
