Index: squid/src/StatHist.c
diff -c squid/src/StatHist.c:1.26 squid/src/StatHist.c:1.26.2.1
*** squid/src/StatHist.c:1.26	Wed Oct 24 02:19:08 2001
--- squid/src/StatHist.c	Thu Sep  1 15:52:23 2005
***************
*** 90,96 ****
      /* a max value should go into the last bin */
      assert(statHistBin(H, max) == H->capacity - 1);
      /* it is hard to test val_out, here is a crude test */
!     assert(((int) floor(0.99L + statHistVal(H, 0) - min)) == 0);
  }
  
  void
--- 90,96 ----
      /* a max value should go into the last bin */
      assert(statHistBin(H, max) == H->capacity - 1);
      /* it is hard to test val_out, here is a crude test */
!     assert(((int) floor(0.99 + statHistVal(H, 0) - min)) == 0);
  }
  
  void
