diff -ur squid-2.6.STABLE19/QUICKSTART squid-2.6.STABLE19.new/QUICKSTART
--- squid-2.6.STABLE19/QUICKSTART	2008-02-10 12:15:14.000000000 +0100
+++ squid-2.6.STABLE19.new/QUICKSTART	2008-03-22 08:51:07.389240901 +0100
@@ -12,10 +12,9 @@
 INSTALL in the same directory), you have to configure the squid.conf
 file. This is the list of the values you *need* to change, because no
 sensible defaults could be defined. Do not touch the other variables
-for now.  We assume you have installed Squid in the default location:
-/usr/local/squid
+for now.
 
-Uncomment and edit the following lines in /usr/local/squid/etc/squid.conf:
+Uncomment and edit the following lines in /etc/squid/squid.conf:
 
 ==============================================================================
 
@@ -26,7 +25,7 @@
     as after startup (typically "nobody" and "nogroup").  Do not use
     "root", for security reasons.
 
-cache_dir ufs /usr/local/squid/var/cache 100 16 256
+cache_dir ufs /var/cache/squid 100 16 256
 
     Add here (first number, here 100) the amount of hard disk space 
     (in megabytes) to devote to caching.
@@ -69,12 +68,12 @@
 line TWICE:
 
 To create any disk cache_dir configured:
-    % /usr/local/squid/sbin/squid -z
+    % /usr/sbin/squid -z
 
 To start squid:
-    % /usr/local/squid/sbin/squid 
+    % /usr/sbin/squid 
 
-Check in the cache.log (/usr/local/squid/var/logs/cache.log) that
+Check in the cache.log (/var/log/squid/cache.log) that
 everything is all right.
 
 Once Squid created all its files (it can take several minutes on some
