diff -Nur squid-2.5.STABLE3.orig/src/client_side.c squid-2.5.STABLE3/src/client_side.c
--- squid-2.5.STABLE3.orig/src/client_side.c	Mon Sep  8 15:44:34 2003
+++ squid-2.5.STABLE3/src/client_side.c	Mon Sep  8 17:00:06 2003
@@ -988,11 +988,16 @@
 #if USE_USERAGENT_LOG
     if ((str = httpHeaderGetStr(req_hdr, HDR_USER_AGENT)))
 	logUserAgent(fqdnFromAddr(http->conn->log_addr), str);
+    else
+	logUserAgent(fqdnFromAddr(http->conn->log_addr), "-");
 #endif
 #if USE_REFERER_LOG
     if ((str = httpHeaderGetStr(req_hdr, HDR_REFERER)))
 	logReferer(fqdnFromAddr(http->conn->log_addr), str,
 	    http->log_uri);
+    else
+	logReferer(fqdnFromAddr(http->conn->log_addr), "-",
+	http->log_uri);
 #endif
 #if FORW_VIA_DB
     if (httpHeaderHas(req_hdr, HDR_X_FORWARDED_FOR)) {
