--- squid.conf	Sun Feb 11 12:02:37 2001
+++ squid.conf.default	Sun Feb 11 12:02:38 2001
@@ -460,6 +460,9 @@
 #	will be created under each first-level directory.  The default
 #	is 256.
 #
+#	NOTE: Remeber to initialize cache directories:
+#	      squid -z -F
+#
 #cache_dir ufs /var/cache/squid 100 16 256
 
 #  TAG: cache_access_log
@@ -467,6 +470,7 @@
 #	every HTTP and ICP queries received.
 #
 #cache_access_log /var/log/squid/access
+cache_access_log /var/log/squid/access.log
 
 #  TAG: cache_log
 #	Cache logging file. This is where general information about
@@ -474,6 +478,7 @@
 #	logged to this file with the "debug_options" tag below.
 #
 #cache_log /var/log/squid/cache
+cache_log /var/log/squid/cache.log
 
 #  TAG: cache_store_log
 #	Logs the activities of the storage manager.  Shows which
@@ -483,6 +488,7 @@
 #	disable it.
 #
 #cache_store_log /var/log/squid/store
+cache_store_log /var/log/squid/store.log
 
 #  TAG: cache_swap_log
 #	Location for the cache "swap.log."  This log file holds the
@@ -1282,6 +1288,8 @@
 #
 #cache_effective_user nobody
 #cache_effective_group nogroup
+cache_effective_user squid 
+cache_effective_group squid
 
 #  TAG: visible_hostname
 #	If you want to present a special hostname in error messages, etc,
@@ -1416,7 +1424,12 @@
 #	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 #	<pid>'.
 #
+#	Default value for logfile_rotate is 10, but PLD uses 'logrotate',
+#	which is more configurable, to perform this task. Setting this
+#	value to '0' disables built-in log rotation and allows 'logrotate'
+#	to do it's job.
 #logfile_rotate 10
+logfile_rotate 0
 
 #  TAG: append_domain
 #	Appends local domain name to hostnames without any dots in
