Index: squid/lib/radix.c
diff -c squid/lib/radix.c:1.16 squid/lib/radix.c:1.16.2.1
*** squid/lib/radix.c:1.16	Tue Nov 13 12:24:34 2001
--- squid/lib/radix.c	Wed Jun 18 17:53:34 2003
***************
*** 113,121 ****
  
  #include "radix.h"
  
  int squid_max_keylen;
  struct squid_radix_mask *squid_rn_mkfreelist;
! struct squid_radix_node_head *squid_mask_rnhead;
  static char *addmask_key;
  static unsigned char normal_chars[] =
  {0, 0x80, 0xc0, 0xe0, 0xf0, 0xf8, 0xfc, 0xfe, 0xFF};
--- 113,127 ----
  
  #include "radix.h"
  
+ 
  int squid_max_keylen;
  struct squid_radix_mask *squid_rn_mkfreelist;
! /* Silly construct to get rid of GCC-3.3 warning about type-punning */
! union {
! 	struct squid_radix_node_head *rn;
! 	void *ptr;
! } squid_mask_rnhead_u;
! #define squid_mask_rnhead squid_mask_rnhead_u.rn
  static char *addmask_key;
  static unsigned char normal_chars[] =
  {0, 0x80, 0xc0, 0xe0, 0xf0, 0xf8, 0xfc, 0xfe, 0xFF};
***************
*** 981,987 ****
      addmask_key = cplim = rn_ones + squid_max_keylen;
      while (cp < cplim)
  	*cp++ = -1;
!     if (squid_rn_inithead((void **) &squid_mask_rnhead, 0) == 0) {
  	fprintf(stderr, "rn_init2 failed.\n");
  	exit(-1);
      }
--- 987,993 ----
      addmask_key = cplim = rn_ones + squid_max_keylen;
      while (cp < cplim)
  	*cp++ = -1;
!     if (squid_rn_inithead(&squid_mask_rnhead_u.ptr, 0) == 0) {
  	fprintf(stderr, "rn_init2 failed.\n");
  	exit(-1);
      }
Index: squid/lib/rfc1738.c
diff -c squid/lib/rfc1738.c:1.23 squid/lib/rfc1738.c:1.23.2.1
*** squid/lib/rfc1738.c:1.23	Wed Feb  7 11:56:51 2001
--- squid/lib/rfc1738.c	Wed Jun 18 17:53:34 2003
***************
*** 128,135 ****
  	    do_escape = 1;
  	}
  	/* RFC 1738 says any non-US-ASCII are encoded */
! 	if (((unsigned char) *p >= (unsigned char) 0x80) &&
! 	    ((unsigned char) *p <= (unsigned char) 0xFF)) {
  	    do_escape = 1;
  	}
  	/* Do the triplet encoding, or just copy the char */
--- 128,134 ----
  	    do_escape = 1;
  	}
  	/* RFC 1738 says any non-US-ASCII are encoded */
! 	if (((unsigned char) *p >= (unsigned char) 0x80)) {
  	    do_escape = 1;
  	}
  	/* Do the triplet encoding, or just copy the char */
Index: squid/src/asn.c
diff -c squid/src/asn.c:1.78 squid/src/asn.c:1.78.2.1
*** squid/src/asn.c:1.78	Tue Nov 13 12:24:35 2001
--- squid/src/asn.c	Wed Jun 18 17:53:34 2003
***************
*** 50,56 ****
  /* END of definitions for radix tree entries */
  
  /* Head for ip to asn radix tree */
! struct squid_radix_node_head *AS_tree_head;
  
  /*
   * Structure for as number information. it could be simply 
--- 50,62 ----
  /* END of definitions for radix tree entries */
  
  /* Head for ip to asn radix tree */
! /* Silly union construct to get rid of GCC-3.3 warning */
! union {
!     struct squid_radix_node_head *rn;
!     void *ptr;
! } AS_tree_head_u;
! 
! #define AS_tree_head AS_tree_head_u.rn
  
  /*
   * Structure for as number information. it could be simply 
***************
*** 160,166 ****
      CBDATA_INIT_TYPE(ASState);
      if (0 == inited++)
  	squid_rn_init();
!     squid_rn_inithead((void **) &AS_tree_head, 8);
      asnAclInitialize(Config.aclList);
      cachemgrRegister("asndb", "AS Number Database", asnStats, 0, 1);
  }
--- 166,172 ----
      CBDATA_INIT_TYPE(ASState);
      if (0 == inited++)
  	squid_rn_init();
!     squid_rn_inithead(&AS_tree_head_u.ptr, 8);
      asnAclInitialize(Config.aclList);
      cachemgrRegister("asndb", "AS Number Database", asnStats, 0, 1);
  }
Index: squid/src/delay_pools.c
diff -c squid/src/delay_pools.c:1.19.2.7 squid/src/delay_pools.c:1.19.2.8
*** squid/src/delay_pools.c:1.19.2.7	Sun Mar  9 11:41:27 2003
--- squid/src/delay_pools.c	Wed Jun 18 17:53:35 2003
***************
*** 397,403 ****
  	position |= 255;
  	if (!(delay_data[pool].class3->individual_255_used[i / 8] & (1 << (i % 8)))) {
  	    delay_data[pool].class3->individual_255_used[i / 8] |= (1 << (i % 8));
- 	    assert(position < C3_IND_SZ);
  	    delay_data[pool].class3->individual[position] =
  		(int) (((double) Config.Delay.rates[pool]->individual.max_bytes *
  		    Config.Delay.initial) / 100);
--- 397,402 ----
***************
*** 415,421 ****
  	    assert(j < (IND_MAP_SZ - 1));
  	    delay_data[pool].class3->individual_map[i][j + 1] = 255;
  	    position |= j;
- 	    assert(position < C3_IND_SZ);
  	    delay_data[pool].class3->individual[position] =
  		(int) (((double) Config.Delay.rates[pool]->individual.max_bytes *
  		    Config.Delay.initial) / 100);
--- 414,419 ----
Index: squid/src/snmp_core.c
diff -c squid/src/snmp_core.c:1.51.2.2 squid/src/snmp_core.c:1.51.2.3
*** squid/src/snmp_core.c:1.51.2.2	Thu Feb 14 12:02:17 2002
--- squid/src/snmp_core.c	Wed Jun 18 17:53:35 2003
***************
*** 607,616 ****
  	    return (Answer);
  	} else if (PDU->command == SNMP_PDU_GETNEXT) {
  	    oid *NextOidName = NULL;
! 	    int NextOidNameLen = 0;
  
  	    ParseFn = snmpTreeNext(PDU->variables->name, PDU->variables->name_length,
! 		&(NextOidName), (snint *) & NextOidNameLen);
  
  	    if (ParseFn == NULL) {
  		Answer->errstat = SNMP_ERR_NOSUCHNAME;
--- 607,616 ----
  	    return (Answer);
  	} else if (PDU->command == SNMP_PDU_GETNEXT) {
  	    oid *NextOidName = NULL;
! 	    snint NextOidNameLen = 0;
  
  	    ParseFn = snmpTreeNext(PDU->variables->name, PDU->variables->name_length,
! 		&NextOidName, &NextOidNameLen);
  
  	    if (ParseFn == NULL) {
  		Answer->errstat = SNMP_ERR_NOSUCHNAME;
