--- squid-2.6.STABLE16/configure.in~	2007-09-19 16:30:59.000000000 +0000
+++ squid-2.6.STABLE16/configure.in	2007-09-19 16:31:03.000000000 +0000
@@ -1085,9 +1085,9 @@
 	    echo "ERROR: Build environment $buildmodel not known to getconf."
 	    exit 1
 	fi
-	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
+	CFLAGS="`getconf ${buildmodel}_CFLAGS | sed -e 's,-m32,,'` $CFLAGS"
 	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
-	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
+	LDFLAGS="`getconf ${buildmodel}_LDFLAGS | sed -e 's,-m32,,'` $LDFLAGS"
 	case "$host" in
 dnl
 dnl On Solaris getconf returns for CFLAGS -Xa and -Usun options, but:
