Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.38 squid/src/cf.data.pre:1.245.2.39
*** squid/src/cf.data.pre:1.245.2.38	Tue Apr  1 21:28:15 2003
--- squid/src/cf.data.pre	Tue Apr  1 21:39:02 2003
***************
*** 274,279 ****
--- 274,281 ----
  		     digest-url=url
  		     allow-miss
  		     max-conn
+ 		     htcp
+ 		     carp-load-factor
  
  		     use 'proxy-only' to specify that objects fetched
  		     from this cache should not be saved locally.
***************
*** 363,369 ****
  		     use 'max-conn' to limit the amount of connections Squid
  		     may open to this peer.
  
! 	NOTE: non-ICP neighbors must be specified as 'parent'.
  DOC_END
  
  NAME: cache_peer_domain cache_host_domain
--- 365,381 ----
  		     use 'max-conn' to limit the amount of connections Squid
  		     may open to this peer.
  
! 		     use 'htcp' to send HTCP, instead of ICP, queries
! 		     to the neighbor.  You probably also want to
! 		     set the "icp port" to 4827 instead of 3130.
! 
! 		     use 'carp-load-factor=f' to define a parent
! 		     cache as one participating in a CARP array.
! 		     The 'f' values for all CARP parents must add
! 		     up to 1.0.
! 		 
! 
! 	NOTE: non-ICP/HTCP neighbors must be specified as 'parent'.
  DOC_END
  
  NAME: cache_peer_domain cache_host_domain
