Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.61 squid/src/cf.data.pre:1.245.2.62
*** squid/src/cf.data.pre:1.245.2.61	Tue Feb 10 14:01:21 2004
--- squid/src/cf.data.pre	Sat Apr  3 06:51:17 2004
***************
*** 862,875 ****
  DOC_END
  
  
! NAME: cache_swap_log
  TYPE: string
  LOC: Config.Log.swap
  DEFAULT: none
  DOC_START
! 	Location for the cache "swap.log."  This log file holds the
! 	metadata of objects saved on disk.  It is used to rebuild the
! 	cache during startup.  Normally this file resides in each
  	'cache_dir' directory, but you may specify an alternate
  	pathname here.  Note you must give a full filename, not just
  	a directory. Since this is the index for the whole object
--- 862,875 ----
  DOC_END
  
  
! NAME: cache_swap_log cache_swap_state
  TYPE: string
  LOC: Config.Log.swap
  DEFAULT: none
  DOC_START
! 	Location for the cache "swap.state" file. This log file holds
! 	the metadata of objects saved on disk.  It is used to rebuild
! 	the cache during startup.  Normally this file resides in each
  	'cache_dir' directory, but you may specify an alternate
  	pathname here.  Note you must give a full filename, not just
  	a directory. Since this is the index for the whole object
