diff -ru squid-2.2.STABLE3.old/configure.in squid-2.2.STABLE3/configure.in
--- squid-2.2.STABLE3.old/configure.in	Wed Apr 21 05:36:01 1999
+++ squid-2.2.STABLE3/configure.in	Sat May 22 10:13:06 1999
@@ -389,7 +389,7 @@
 [  if test "$enableval" = "yes" ; then
      echo "ARP ACL lists enabled (ether address)"
      case "$host" in
-	*-pc-linux-*)
+	*-linux-*)
 	    ;;
 	*)
 	    echo "WARNING: ARP ACL support is only currently supported on Linux"
Only in squid-2.2.STABLE3: configure.in~
