Index: squid/src/fqdncache.c
diff -c squid/src/fqdncache.c:1.149 squid/src/fqdncache.c:1.149.2.1
*** squid/src/fqdncache.c:1.149	Wed Oct 24 00:16:16 2001
--- squid/src/fqdncache.c	Sat Oct 11 16:32:31 2003
***************
*** 278,284 ****
      int k;
      int na = 0;
      memset(&f, '\0', sizeof(f));
!     f.expires = squid_curtime;
      f.flags.negcached = 1;
      if (nr < 0) {
  	debug(35, 3) ("fqdncacheParse: Lookup failed (error %d)\n",
--- 278,284 ----
      int k;
      int na = 0;
      memset(&f, '\0', sizeof(f));
!     f.expires = squid_curtime + Config.negativeDnsTtl;
      f.flags.negcached = 1;
      if (nr < 0) {
  	debug(35, 3) ("fqdncacheParse: Lookup failed (error %d)\n",
