Index: src/client_side.c
===================================================================
RCS file: /server/cvs-server/squid/squid/src/client_side.c,v
retrieving revision 1.459.2.45
retrieving revision 1.459.2.46
diff -c -r1.459.2.45 -r1.459.2.46
*** src/client_side.c	2000/07/18 02:28:33	1.459.2.45
--- src/client_side.c	2000/09/07 04:06:59	1.459.2.46
***************
*** 145,151 ****
      clientHttpRequest *http = data;
      ConnStateData *conn = http->conn;
      if (checkAccelOnly(http)) {
! 	clientAccessCheckDone(ACCESS_ALLOWED, http);
  	return;
      }
      http->acl_checklist = aclChecklistCreate(Config.accessList.http,
--- 145,152 ----
      clientHttpRequest *http = data;
      ConnStateData *conn = http->conn;
      if (checkAccelOnly(http)) {
! 	/* deny proxy requests in accel_only mode */
! 	clientAccessCheckDone(ACCESS_DENIED, http);
  	return;
      }
      http->acl_checklist = aclChecklistCreate(Config.accessList.http,
