Index: squid/src/cf.data.pre
diff -c squid/src/cf.data.pre:1.245.2.45 squid/src/cf.data.pre:1.245.2.46
*** squid/src/cf.data.pre:1.245.2.45	Tue Jun 10 14:45:58 2003
--- squid/src/cf.data.pre	Thu Jun 19 15:29:48 2003
***************
*** 416,422 ****
  DEFAULT: none
  LOC: none
  DOC_START
! 	usage: neighbor_type_domain parent|sibling domain domain ...
  
  	Modifying the neighbor type for specific domains is now
  	possible.  You can treat some domains differently than the the
--- 416,422 ----
  DEFAULT: none
  LOC: none
  DOC_START
! 	usage: neighbor_type_domain neighbor parent|sibling domain domain ...
  
  	Modifying the neighbor type for specific domains is now
  	possible.  You can treat some domains differently than the the
