Index: squid/src/main.c
diff -c squid/src/main.c:1.345.2.11 squid/src/main.c:1.345.2.12
*** squid/src/main.c:1.345.2.11	Sun Jun  8 17:28:46 2003
--- squid/src/main.c	Fri Sep 19 00:36:59 2003
***************
*** 329,335 ****
  static void
  mainReconfigure(void)
  {
!     debug(1, 1) ("Restarting Squid Cache (version %s)...\n", version_string);
      reconfiguring = 1;
      /* Already called serverConnectionsClose and ipcacheShutdownServers() */
      serverConnectionsClose();
--- 329,335 ----
  static void
  mainReconfigure(void)
  {
!     debug(1, 1) ("Reconfiguring Squid Cache (version %s)...\n", version_string);
      reconfiguring = 1;
      /* Already called serverConnectionsClose and ipcacheShutdownServers() */
      serverConnectionsClose();
