Index: squid3/src/Debug.h
diff -c squid3/src/Debug.h:1.6 squid3/src/Debug.h:1.7
*** squid3/src/Debug.h:1.6	Tue Jul 15 23:27:17 2003
--- squid3/src/Debug.h	Sun Aug 31 15:20:08 2003
***************
*** 60,65 ****
--- 60,66 ----
      static int level;
      static std::ostream &getDebugOut();
      static void finishDebug();
+     static void parseOptions(char const *);
  
  private:
      static std::ostringstream *CurrentDebug;
Index: squid3/src/cache_cf.cc
diff -c squid3/src/cache_cf.cc:1.448 squid3/src/cache_cf.cc:1.449
*** squid3/src/cache_cf.cc:1.448	Wed Aug 13 10:05:22 2003
--- squid3/src/cache_cf.cc	Sun Aug 31 15:20:08 2003
***************
*** 2014,2019 ****
--- 2014,2021 ----
  #define free_onoff free_int
  #define dump_eol dump_string
  #define free_eol free_string
+ #define dump_debug dump_string
+ #define free_debug free_string
  
  static void
  dump_refreshpattern(StoreEntry * entry, const char *name, refresh_t * head)
***************
*** 2232,2237 ****
--- 2234,2248 ----
          self_destruct();
  
      *var = xstrdup((char *) token);
+ }
+ 
+ void
+ parse_debug(char *volatile *var)
+ {
+     parse_eol(var);
+     safe_free(debug_options)
+     debug_options = xstrdup(Config.debugOptions);
+     Debug::parseOptions(Config.debugOptions);
  }
  
  static void
Index: squid3/src/debug.cc
diff -c squid3/src/debug.cc:1.95 squid3/src/debug.cc:1.96
*** squid3/src/debug.cc:1.95	Mon Jul 14 02:21:56 2003
--- squid3/src/debug.cc	Sun Aug 31 15:20:08 2003
***************
*** 232,238 ****
  }
  
  void
! _db_init(const char *logfile, const char *options) {
      int i;
      char *p = NULL;
      char *s = NULL;
--- 232,238 ----
  }
  
  void
! Debug::parseOptions(char const *options) {
      int i;
      char *p = NULL;
      char *s = NULL;
***************
*** 248,253 ****
--- 248,258 ----
  
          xfree(p);
      }
+ }
+ 
+ void
+ _db_init(const char *logfile, const char *options) {
+     Debug::parseOptions(options);
  
      debugOpenLog(logfile);
  
Index: squid3/src/cf.data.pre
diff -c squid3/src/cf.data.pre:1.340 squid3/src/cf.data.pre:1.341
*** squid3/src/cf.data.pre:1.340	Wed Aug 27 15:19:32 2003
--- squid3/src/cf.data.pre	Sun Aug 31 15:20:08 2003
***************
*** 1323,1329 ****
  
  
  NAME: debug_options
! TYPE: eol
  DEFAULT: ALL,1
  LOC: Config.debugOptions
  DOC_START
--- 1323,1329 ----
  
  
  NAME: debug_options
! TYPE: debug
  DEFAULT: ALL,1
  LOC: Config.debugOptions
  DOC_START
