Index: squid/src/main.c
diff -c squid/src/main.c:1.345.2.12 squid/src/main.c:1.345.2.13
*** squid/src/main.c:1.345.2.12	Fri Sep 19 00:36:59 2003
--- squid/src/main.c	Wed Dec 17 14:10:30 2003
***************
*** 832,837 ****
--- 832,841 ----
  {
      pid_t pid;
      debug_log = stderr;
+     if (strcmp(Config.pidFilename, "none") == 0) {
+ 	debug(0, 1) ("No pid_filename specified. Trusting you know what you are doing.\n");
+ 	return 0;
+     }
      pid = readPidFile();
      if (pid < 2)
  	return 0;
