Index: squid/src/external_acl.c
diff -c squid/src/external_acl.c:1.1.2.21 squid/src/external_acl.c:1.1.2.22
*** squid/src/external_acl.c:1.1.2.21	Mon Mar 24 10:25:51 2003
--- squid/src/external_acl.c	Wed Apr  9 06:35:25 2003
***************
*** 433,439 ****
      }
      if (!entry || entry->result == -1) {
  	debug(82, 2) ("aclMatchExternal: %s(\"%s\") = lookup needed\n", acl->def->name, key);
! 	ch->state[ACL_EXTERNAL] = ACL_LOOKUP_NEEDED;
  	return 0;
      }
      external_acl_cache_touch(acl->def, entry);
--- 433,442 ----
      }
      if (!entry || entry->result == -1) {
  	debug(82, 2) ("aclMatchExternal: %s(\"%s\") = lookup needed\n", acl->def->name, key);
! 	if (acl->def->helper->stats.queue_size >= acl->def->helper->n_running)
! 	    debug(82, 1) ("aclMatchExternal: '%s' queue overload. Request rejected.\n", acl->def->name);
! 	else
! 	    ch->state[ACL_EXTERNAL] = ACL_LOOKUP_NEEDED;
  	return 0;
      }
      external_acl_cache_touch(acl->def, entry);
