diff -urN squid-2.6.STABLE5.orig/src/acl.c squid-2.6.STABLE5/src/acl.c
--- squid-2.6.STABLE5.orig/src/acl.c	2006-10-16 22:11:41.000000000 +0200
+++ squid-2.6.STABLE5/src/acl.c	2006-11-28 18:10:47.089402554 +0100
@@ -843,7 +843,7 @@
     while ((t = strtokFile())) {
 	Tolower(t);
 	t = xstrdup(t);
-	*Top = splay_insert(t, *Top, aclDomainCompare);
+	*Top = splay_insert(t, *Top, aclHostDomainCompare);
 	if (splayLastResult == 0)
 	    safe_free(t);
     }
@@ -869,7 +869,7 @@
     Top = &(*datap)->values;
     while ((t = strtokFile())) {
 	t = xstrdup(t);
-	*Top = splay_insert(t, *Top, aclDomainCompare);
+	*Top = splay_insert(t, *Top, aclHostDomainCompare);
 	if (splayLastResult == 0)
 	    safe_free(t);
     }
