--- squid-3.0.STABLE4/configure.in	2008-04-01 13:54:39.000000000 +0200
+++ squid-3.0.STABLE4.new/configure.in	2008-04-20 15:28:17.454985027 +0200
@@ -1186,10 +1186,10 @@
 	    echo "ERROR: Build environment $buildmodel not known to getconf."
 	    exit 1
 	fi
-	CFLAGS="`getconf ${buildmodel}_CFLAGS` $CFLAGS"
-	CXXFLAGS="`getconf ${buildmodel}_CFLAGS` $CXXFLAGS"
+	CFLAGS="`getconf ${buildmodel}_CFLAGS` | sed -e 's,-m32,,'` $CFLAGS"
+	CXXFLAGS="`getconf ${buildmodel}_CFLAGS` | sed -e 's,-m32,,'` $CXXFLAGS"
 	LIBS="`getconf ${buildmodel}_LIBS` $LIBS"
-	LDFLAGS="`getconf ${buildmodel}_LDFLAGS` $LDFLAGS"
+	LDFLAGS="`getconf ${buildmodel}_LDFLAGS | sed -e 's,-m32,,'` $LDFLAGS"
 	case "$host" in
 dnl
 dnl On Solaris getconf returns for CFLAGS -Xa and -Usun options, but:
