Index: squid3/aclocal.m4
diff -c squid3/aclocal.m4:1.35 squid3/aclocal.m4:1.36
*** squid3/aclocal.m4:1.35	Mon Aug 18 18:20:25 2003
--- squid3/aclocal.m4	Sat Aug 30 03:14:37 2003
***************
*** 46,51 ****
--- 46,54 ----
  dnl *BSD net headers
  AC_DEFUN(SQUID_BSDNET_INCLUDES,[
  SQUID_DEFAULT_INCLUDES
+ #if HAVE_SYS_TIME_H
+ #include <sys/time.h>
+ #endif
  #if HAVE_SYS_SOCKET_H
  #include <sys/socket.h>
  #endif
***************
*** 55,65 ****
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
- #if HAVE_NETINET_IP_FIL_H
- #include <netinet/ip_fil.h>
- #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
--- 58,68 ----
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
+ #endif
+ #if HAVE_NETINET_IP_FIL_H
+ #include <netinet/ip_fil.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
Index: squid3/configure
diff -c squid3/configure:1.368 squid3/configure:1.369
*** squid3/configure:1.368	Mon Aug 18 18:20:25 2003
--- squid3/configure	Sat Aug 30 03:14:37 2003
***************
*** 7782,7787 ****
--- 7782,7790 ----
  #include <sys/select.h>
  #endif
  
+ #if HAVE_SYS_TIME_H
+ #include <sys/time.h>
+ #endif
  #if HAVE_SYS_SOCKET_H
  #include <sys/socket.h>
  #endif
***************
*** 7791,7801 ****
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
- #if HAVE_NETINET_IP_FIL_H
- #include <netinet/ip_fil.h>
- #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
--- 7794,7804 ----
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
+ #endif
+ #if HAVE_NETINET_IP_FIL_H
+ #include <netinet/ip_fil.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
Index: squid3/acinclude.m4
diff -c squid3/acinclude.m4:1.9 squid3/acinclude.m4:1.10
*** squid3/acinclude.m4:1.9	Mon Aug 18 18:15:01 2003
--- squid3/acinclude.m4	Sat Aug 30 03:08:21 2003
***************
*** 33,38 ****
--- 33,41 ----
  dnl *BSD net headers
  AC_DEFUN(SQUID_BSDNET_INCLUDES,[
  SQUID_DEFAULT_INCLUDES
+ #if HAVE_SYS_TIME_H
+ #include <sys/time.h>
+ #endif
  #if HAVE_SYS_SOCKET_H
  #include <sys/socket.h>
  #endif
***************
*** 42,52 ****
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
- #if HAVE_NETINET_IP_FIL_H
- #include <netinet/ip_fil.h>
- #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
--- 45,55 ----
  #if HAVE_NETINET_IP_COMPAT_H
  #include <netinet/ip_compat.h>
  #endif
  #if HAVE_NET_IF_H
  #include <net/if.h>
+ #endif
+ #if HAVE_NETINET_IP_FIL_H
+ #include <netinet/ip_fil.h>
  #endif
  #if HAVE_SYS_PARAM_H
  #include <sys/param.h>
