Index: squid/src/tools.c
diff -c squid/src/tools.c:1.213.2.20 squid/src/tools.c:1.213.2.22
*** squid/src/tools.c:1.213.2.20	Fri Sep 16 15:13:18 2005
--- squid/src/tools.c	Sat Jan 14 18:17:47 2006
***************
*** 563,569 ****
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
--- 563,569 ----
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1, 0, 0, 0) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
***************
*** 580,586 ****
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
--- 580,586 ----
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1, 0, 0, 0) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
***************
*** 605,611 ****
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
--- 605,611 ----
  #endif
  #if HAVE_PRCTL && defined(PR_SET_DUMPABLE)
      /* Set Linux DUMPABLE flag */
!     if (Config.coredump_dir && prctl(PR_SET_DUMPABLE, 1, 0, 0, 0) != 0)
  	debug(50, 2) ("prctl: %s\n", xstrerror());
  #endif
  }
