Index: squid/configure.in
diff -c squid/configure.in:1.251.2.36 squid/configure.in:1.251.2.39
*** squid/configure.in:1.251.2.36	Sun Sep 29 06:56:26 2002
--- squid/configure.in	Tue Oct 22 02:34:46 2002
***************
*** 263,274 ****
--- 263,276 ----
                            --enable-storeio=ufs,aufs],
  [ case $enableval in
    yes)
+ 	with_pthreads="yes"
  	STORE_MODULES="ufs aufs"
      ;;
    no)
      ;;
    *)
    	aufs_io_threads=$enableval
+ 	with_pthreads="yes"
  	STORE_MODULES="ufs aufs"
      ;;
    esac
***************
*** 288,310 ****
  [  --with-pthreads         Use POSIX Threads])
  if test "$with_pthreads" = "yes"; then
      echo "With pthreads"
-     CFLAGS="$CFLAGS -D_REENTRANT"
-     case "$host" in
-     i386-unknown-freebsd*)
-         if test "$GCC" = "yes" ; then
-             if test -z "$PRESET_LDFLAGS"; then
-                 LDFLAGS="$LDFLAGS -pthread"
-             fi
-         fi
-     ;;
-     *-solaris2.*)
-         if test "$GCC" = "yes" ; then
- 	    CFLAGS="$CFLAGS -pthreads"
- 	else
- 	    CFLAGS="$CFLAGS -mt"
-         fi
-     ;;
-     esac
  fi
  
  AC_ARG_WITH(aio,
--- 290,295 ----
***************
*** 374,380 ****
  AC_SUBST(STORE_MODULES)
  AC_SUBST(STORE_MODULE_SUBDIRS)
  
- 
  dnl --enable-heap-replacement compability option
  AC_ARG_ENABLE(heap-replacement,
  [  --enable-heap-replacement
--- 359,364 ----
***************
*** 1482,1487 ****
--- 1466,1488 ----
  dnl Check for pthreads
  dnl We use pthreads when doing ASYNC I/O
  if test "$with_pthreads" = "yes"; then
+     CFLAGS="$CFLAGS -D_REENTRANT"
+     case "$host" in
+     i386-unknown-freebsd*)
+         if test "$GCC" = "yes" ; then
+             if test -z "$PRESET_LDFLAGS"; then
+                 LDFLAGS="$LDFLAGS -pthread"
+             fi
+         fi
+     ;;
+     *-solaris2.*)
+         if test "$GCC" = "yes" ; then
+ 	    CFLAGS="$CFLAGS -pthreads"
+ 	else
+ 	    CFLAGS="$CFLAGS -mt"
+         fi
+     ;;
+     esac
      AC_CHECK_LIB(pthread, main)
  fi
  
Index: squid/configure
diff -c squid/configure:1.248.2.36 squid/configure:1.248.2.40
*** squid/configure:1.248.2.36	Thu Oct  3 09:41:57 2002
--- squid/configure	Tue Oct 22 02:40:11 2002
***************
*** 1856,1867 ****
--- 1856,1869 ----
    enableval="$enable_async_io"
     case $enableval in
    yes)
+ 	with_pthreads="yes"
  	STORE_MODULES="ufs aufs"
      ;;
    no)
      ;;
    *)
    	aufs_io_threads=$enableval
+ 	with_pthreads="yes"
  	STORE_MODULES="ufs aufs"
      ;;
    esac
***************
*** 1891,1913 ****
  
  if test "$with_pthreads" = "yes"; then
      echo "With pthreads"
-     CFLAGS="$CFLAGS -D_REENTRANT"
-     case "$host" in
-     i386-unknown-freebsd*)
-         if test "$GCC" = "yes" ; then
-             if test -z "$PRESET_LDFLAGS"; then
-                 LDFLAGS="$LDFLAGS -pthread"
-             fi
-         fi
-     ;;
-     *-solaris2.*)
-         if test "$GCC" = "yes" ; then
- 	    CFLAGS="$CFLAGS -pthreads"
- 	else
- 	    CFLAGS="$CFLAGS -mt"
-         fi
-     ;;
-     esac
  fi
  
  # Check whether --with-aio or --without-aio was given.
--- 1967,1972 ----
***************
*** 7066,7071 ****
  fi
  
  if test "$with_pthreads" = "yes"; then
      echo $ac_n "checking for main in -lpthread""... $ac_c" 1>&6
  echo "configure:7071: checking for main in -lpthread" >&5
  ac_lib_var=`echo pthread'_'main | sed 'y%./+-%__p_%'`
--- 7050,7074 ----
  fi
  
  if test "$with_pthreads" = "yes"; then
+     CFLAGS="$CFLAGS -D_REENTRANT"
+     case "$host" in
+     i386-unknown-freebsd*)
+         if test "$GCC" = "yes" ; then
+             if test -z "$PRESET_LDFLAGS"; then
+                 LDFLAGS="$LDFLAGS -pthread"
+             fi
+         fi
+     ;;
+     *-solaris2.*)
+         if test "$GCC" = "yes" ; then
+ 	    CFLAGS="$CFLAGS -pthreads"
+ 	else
+ 	    CFLAGS="$CFLAGS -mt"
+         fi
+     ;;
+     esac
      echo $ac_n "checking for main in -lpthread""... $ac_c" 1>&6
  echo "configure:7072: checking for main in -lpthread" >&5
  ac_lib_var=`echo pthread'_'main | sed 'y%./+-%__p_%'`
