Index: squid/src/auth/digest/auth_digest.c
diff -c squid/src/auth/digest/auth_digest.c:1.10.2.12 squid/src/auth/digest/auth_digest.c:1.10.2.13
*** squid/src/auth/digest/auth_digest.c:1.10.2.12	Sat Apr 17 19:29:52 2004
--- squid/src/auth/digest/auth_digest.c	Sat Aug 28 16:31:15 2004
***************
*** 893,899 ****
      if (reply) {
  	if ((t = strchr(reply, ' ')))
  	    *t = '\0';
! 	if (*reply == '\0')
  	    reply = NULL;
      }
      assert(r->auth_user_request != NULL);
--- 893,899 ----
      if (reply) {
  	if ((t = strchr(reply, ' ')))
  	    *t = '\0';
! 	if (*reply == '\0' || *reply == '\n')
  	    reply = NULL;
      }
      assert(r->auth_user_request != NULL);
***************
*** 903,909 ****
      digest_user = auth_user_request->auth_user->scheme_data;
      if (reply && (strncasecmp(reply, "ERR", 3) == 0))
  	digest_request->flags.credentials_ok = 3;
!     else {
  	CvtBin(reply, digest_user->HA1);
  	digest_user->HA1created = 1;
      }
--- 903,909 ----
      digest_user = auth_user_request->auth_user->scheme_data;
      if (reply && (strncasecmp(reply, "ERR", 3) == 0))
  	digest_request->flags.credentials_ok = 3;
!     else if (reply) {
  	CvtBin(reply, digest_user->HA1);
  	digest_user->HA1created = 1;
      }
