Index: squid/src/ftp.c
diff -c squid/src/ftp.c:1.316.2.22 squid/src/ftp.c:1.316.2.23
*** squid/src/ftp.c:1.316.2.22	Fri Mar  4 04:19:53 2005
--- squid/src/ftp.c	Fri Mar  4 04:53:55 2005
***************
*** 1648,1653 ****
--- 1648,1654 ----
  	debug(9, 3) ("Directory path did not end in /\n");
  	strCat(ftpState->title_url, "/");
  	ftpState->flags.isdir = 1;
+ 	ftpState->flags.use_base = 1;
      }
      ftpSendPasv(ftpState);
  }
***************
*** 2123,2129 ****
  ftpSendList(FtpStateData * ftpState)
  {
      if (ftpState->filepath) {
- 	ftpState->flags.use_base = 1;
  	snprintf(cbuf, 1024, "LIST %s\r\n", ftpState->filepath);
      } else {
  	snprintf(cbuf, 1024, "LIST\r\n");
--- 2124,2129 ----
