Index: squid/src/helper.c
diff -c squid/src/helper.c:1.34.2.15 squid/src/helper.c:1.34.2.16
*** squid/src/helper.c:1.34.2.15	Mon Feb  9 02:03:49 2004
--- squid/src/helper.c	Thu Mar 11 08:26:30 2004
***************
*** 743,761 ****
      hlp->stats.queue_size++;
      if (hlp->stats.queue_size < hlp->n_running)
  	return;
!     if (hlp->stats.queue_size < hlp->n_running)
! 	return;
      if (squid_curtime - hlp->last_queue_warn < 30)
  	return;
      if (shutting_down || reconfiguring)
  	return;
-     hlp->last_queue_warn = squid_curtime;
      debug(84, 1) ("WARNING: All %s processes are busy.\n", hlp->id_name);
      debug(84, 1) ("WARNING: up to %d pending requests queued\n", hlp->stats.max_queue_size);
      if (hlp->stats.queue_size > hlp->n_running * 2)
  	fatalf("Too many queued %s requests (%d on %d)", hlp->id_name, hlp->stats.queue_size, hlp->n_running);
!     if (hlp->stats.max_queue_size > 1)
  	debug(84, 1) ("Consider increasing the number of %s processes to at least %d in your config file.\n", hlp->id_name, hlp->n_running + hlp->stats.max_queue_size);
      hlp->stats.max_queue_size = hlp->stats.queue_size;
  }
  
--- 743,761 ----
      hlp->stats.queue_size++;
      if (hlp->stats.queue_size < hlp->n_running)
  	return;
!     if (hlp->stats.queue_size > hlp->stats.max_queue_size)
! 	hlp->stats.max_queue_size = hlp->stats.queue_size;
      if (squid_curtime - hlp->last_queue_warn < 30)
  	return;
      if (shutting_down || reconfiguring)
  	return;
      debug(84, 1) ("WARNING: All %s processes are busy.\n", hlp->id_name);
      debug(84, 1) ("WARNING: up to %d pending requests queued\n", hlp->stats.max_queue_size);
      if (hlp->stats.queue_size > hlp->n_running * 2)
  	fatalf("Too many queued %s requests (%d on %d)", hlp->id_name, hlp->stats.queue_size, hlp->n_running);
!     if (squid_curtime - hlp->last_queue_warn < 300)
  	debug(84, 1) ("Consider increasing the number of %s processes to at least %d in your config file.\n", hlp->id_name, hlp->n_running + hlp->stats.max_queue_size);
+     hlp->last_queue_warn = squid_curtime;
      hlp->stats.max_queue_size = hlp->stats.queue_size;
  }
  
***************
*** 775,785 ****
  	return;
      if (shutting_down || reconfiguring)
  	return;
-     hlp->last_queue_warn = squid_curtime;
      debug(84, 1) ("WARNING: All %s processes are busy.\n", hlp->id_name);
      debug(84, 1) ("WARNING: up to %d pending requests queued\n", hlp->stats.max_queue_size);
!     if (hlp->stats.max_queue_size > 1)
  	debug(84, 1) ("Consider increasing the number of %s processes to at least %d in your config file.\n", hlp->id_name, hlp->n_running + hlp->stats.max_queue_size);
      hlp->stats.max_queue_size = hlp->stats.queue_size;
  }
  
--- 775,785 ----
  	return;
      if (shutting_down || reconfiguring)
  	return;
      debug(84, 1) ("WARNING: All %s processes are busy.\n", hlp->id_name);
      debug(84, 1) ("WARNING: up to %d pending requests queued\n", hlp->stats.max_queue_size);
!     if (squid_curtime - hlp->last_queue_warn < 300)
  	debug(84, 1) ("Consider increasing the number of %s processes to at least %d in your config file.\n", hlp->id_name, hlp->n_running + hlp->stats.max_queue_size);
+     hlp->last_queue_warn = squid_curtime;
      hlp->stats.max_queue_size = hlp->stats.queue_size;
  }
  
